<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • biomartr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.6.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.6.9000" id="biomartr-1069000">biomartr 1.0.6.9000<a class="anchor" aria-label="anchor" href="#biomartr-1069000"></a></h2>
<div class="section level4">
<h4 id="new-features-1-0-6-9000">New features<a class="anchor" aria-label="anchor" href="#new-features-1-0-6-9000"></a></h4>
<ul><li><p>Some cool new generalization, and check out function <code>biomartr:::supported_biotypes(db = "refseq")</code>. This function will simplify a lot of stuff downstream. (<a href="https://github.com/ropensci/biomartr/issues/104" class="external-link">#104</a>)</p></li>
<li><p>Tests are now much quicker to run, because <code><a href="../reference/is.genome.available.html">biomartr::is.genome.available</a></code> (which is used basically everywhere) now reads files with data.table instead of reader. (<a href="https://github.com/ropensci/biomartr/issues/104" class="external-link">#104</a>)</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.5" id="biomartr-105"><a href="https://github.com/ropensci/biomartr/releases/tag/v1.0.5" class="external-link">biomartr 1.0.5</a><small>2023-10-04</small><a class="anchor" aria-label="anchor" href="#biomartr-105"></a></h2>
<div class="section level4">
<h4 id="package-generalization-1-0-5">Package generalization<a class="anchor" aria-label="anchor" href="#package-generalization-1-0-5"></a></h4>
<p>Over 5000 lines have been edited, most of them removed (<a href="https://github.com/ropensci/biomartr/issues/100" class="external-link">#100</a>), to generalize the package to make it more safe for future development. This progress is still ongoing.</p>
</div>
<div class="section level4">
<h4 id="new-features-1-0-5">New features<a class="anchor" aria-label="anchor" href="#new-features-1-0-5"></a></h4>
<ul><li>Ensembl genomes is no longer a different database compared to ensembl in biomaRt, since this split is artifical. It is adviced to use only “ensembl” as db from now on, but “ensemblgenomes” will still work.</li>
<li>Annotation did mean gff, but it should be both gff and gtf getter, with format specification, this is now fixed and generalized.</li>
<li>Added in new kingdom for ensembl: protists supportwith correct collection getters</li>
<li>The retrieval from the <code>UniProt</code> database is now updated to the new API/FTP path system. Now users can retrieve proteomes using the functions <code>getProteome(db = "uniprot", ...)</code> and <code>getProteomeSet(db = "uniprot", ...)</code> (see <a href="https://github.com/ropensci/biomartr/issues/82" class="external-link">#82</a>)</li>
<li>new function <code>getBioSet</code>: Generic Bio data set extractor</li>
<li>new function <code>getBio</code>: A wrapper to all bio getters, selected with ‘type’ argument</li>
<li>a new function <code><a href="../reference/getUniProtSTATS.html">getUniProtSTATS()</a></code>: Retrieve UniProt Database Information File (STATS)</li>
</ul></div>
<div class="section level4">
<h4 id="power-user-cache-1-0-5">Power user cache<a class="anchor" aria-label="anchor" href="#power-user-cache-1-0-5"></a></h4>
<p>The package now supports caching of back end files which used to be saved to /tmp folder (i.e. lost on computer restart). This make it easy for power users who want higher speed. For more info, see the function ?cachedir_set</p>
</div>
<div class="section level4">
<h4 id="bug-fixes-1-0-5">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-0-5"></a></h4>
<ul><li>Fixed many wrong urls and non working functions, more tests are added to make sure they work.</li>
<li>Fixed fungi collection accessor for ensembl</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.4" id="biomartr-104"><a href="https://github.com/ropensci/biomartr/releases/tag/v1.0.4" class="external-link">biomartr 1.0.4</a><small>2023-06-20</small><a class="anchor" aria-label="anchor" href="#biomartr-104"></a></h2>
<div class="section level4">
<h4 id="new-features-1-0-4">New Features<a class="anchor" aria-label="anchor" href="#new-features-1-0-4"></a></h4>
<ul><li>in <code><a href="../reference/getSummaryFile.html">getSummaryFile()</a></code> all columns of the <code>assembly_summary.txt</code> are now specified with names and correct data types (<a href="https://github.com/ropensci/biomartr/issues/92" class="external-link">#92</a>)</li>
<li>all <code>get*()</code> functions, the <code><a href="../reference/getKingdomAssemblySummary.html">getKingdomAssemblySummary()</a></code>, and <code>is.genome.available.refseq.genbank()</code> all receive a new argument <code>skip_bacteria</code> which is set to <code>TRUE</code> by default. This ensures that the huge dataset file for <code>bacteria</code> is not downloaded by default when retrieving summary files from <code>GenBank</code>. Users who wish to retrieved data from particular bacteria can actively set <code>skip_bacteria = TRUE</code> in all <code>get*()</code> functions.</li>
</ul></div>
<div class="section level4">
<h4 id="bug-fixes-1-0-4">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-0-4"></a></h4>
<ul><li>whenever the low-level function <code><a href="../reference/getKingdomAssemblySummary.html">getKingdomAssemblySummary()</a></code> was called by all get*() functions, due to an error in the <code>assembly_summary.txt</code> file for viruses where the total gene count was stored as character and not as integer (as is the case for all other <code>assembly_summary.txt</code> files), an error occurred stating that <code><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">dplyr::bind_rows()</a></code> cannot join column $X35 due to differences in data types. This has now been resolved by parsing the correct data types with <code>readr</code>. Many thanks to … for pointing this out to me. (<a href="https://github.com/ropensci/biomartr/issues/92" class="external-link">#92</a>)</li>
<li>fixing md5checks in all <code>get*()</code> functions</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.3" id="biomartr-103"><a href="https://github.com/ropensci/biomartr/releases/tag/v1.0.3" class="external-link">biomartr 1.0.3</a><small>2023-05-07</small><a class="anchor" aria-label="anchor" href="#biomartr-103"></a></h2>
<ul><li>adding <a href="https://github.com/ropensci/biomartr/pull/88" class="external-link">pull request </a><a href="https://github.com/ropensci/biomartr/issues/88" class="external-link">#88</a> which fixes issues with <code>http</code> to <code>https</code> curl requests (Many thanks to <a href="https://github.com/Roleren" class="external-link">@Roleren</a>)</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.0.2" id="biomartr-102"><a href="https://github.com/ropensci/biomartr/releases/tag/v1.0.2" class="external-link">biomartr 1.0.2</a><small>2022-02-23</small><a class="anchor" aria-label="anchor" href="#biomartr-102"></a></h2>
<div class="section level4">
<h4 id="new-functions-1-0-2">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-1-0-2"></a></h4>
<ul><li><p>New function <code><a href="../reference/check_annotation_biomartr.html">check_annotation_biomartr()</a></code> helps to check whether downloaded GFF or GTF files are corrupt. Find more details <a href="https://github.com/lawremi/rtracklayer/issues/15" class="external-link">here</a></p></li>
<li><p>new function <code><a href="../reference/getCollectionSet.html">getCollectionSet()</a></code> allows users to retrieve a Collection: Genome, Proteome, CDS, RNA, GFF, Repeat Masker, AssemblyStats of multiple species</p></li>
</ul><p>Example:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># define scientific names of species for which</span></span>
<span><span class="co"># collections shall be retrieved</span></span>
<span><span class="va">organism_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Arabidopsis thaliana"</span>, </span>
<span>                   <span class="st">"Arabidopsis lyrata"</span>, </span>
<span>                   <span class="st">"Capsella rubella"</span><span class="op">)</span></span>
<span><span class="co"># download the collection of Arabidopsis thaliana from refseq</span></span>
<span><span class="co"># and store the corresponding genome file in '_ncbi_downloads/collection'</span></span>
<span> <span class="fu"><a href="../reference/getCollectionSet.html">getCollectionSet</a></span><span class="op">(</span> db       <span class="op">=</span> <span class="st">"refseq"</span>, </span>
<span>             organism <span class="op">=</span> <span class="va">organism_list</span>, </span>
<span>             path <span class="op">=</span> <span class="st">"set_collections"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="new-features-1-0-2">New Features<a class="anchor" aria-label="anchor" href="#new-features-1-0-2"></a></h4>
<ul><li>the <code><a href="../reference/getGFF.html">getGFF()</a></code> function receives a new argument <code>remove_annotation_outliers</code> to enable users to remove corrupt lines from a GFF file Example:</li>
</ul><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">Ath_path</span> <span class="op">&lt;-</span> <span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/getGFF.html">getGFF</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="st">"Arabidopsis thaliana"</span>, remove_annotation_outliers <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<ul><li><p>the <code><a href="../reference/getGFFSet.html">getGFFSet()</a></code> function receives a new argument <code>remove_annotation_outliers</code> to enable users to remove corrupt lines from a GFF file</p></li>
<li><p>the <code><a href="../reference/getGTF.html">getGTF()</a></code> function receives a new argument <code>remove_annotation_outliers</code> to enable users to remove corrupt lines from a GTF file</p></li>
<li><p>adding a new message system to <code><a href="../reference/organismBM.html">biomartr::organismBM()</a></code>, <code><a href="../reference/organismAttributes.html">biomartr::organismAttributes()</a></code>, and <code><a href="../reference/organismFilters.html">biomartr::organismFilters()</a></code> so that large API queries don’t seem so unresponsive</p></li>
<li><p><code><a href="../reference/getCollection.html">getCollection()</a></code> receives new arguments <code>release</code>, <code>remove_annotation_outliers</code>, and <code>gunzip</code> that will now be passed on to downstream retrieval functions</p></li>
<li><p>the <code><a href="../reference/getGTF.html">getGTF()</a></code>, <code><a href="../reference/getGenome.html">getGenome()</a></code> and <code><a href="../reference/getGenomeSet.html">getGenomeSet()</a></code> functions receives a new argument <code>assembly_type = "toplevel"</code> to enable users to choose between toplevel and primary assembly when using ensembl database. Setting <code>assembly_type = "primary_assembly"</code> will save a lot a space on hard drives for people using large ensembl genomes.</p></li>
<li><p>all <code>get*()</code> functions with <code>release</code> argument now check if the ENSEMBL release is &gt;45 (Many thanks to <a href="https://github.com/Roleren" class="external-link">@Roleren</a> <a href="https://github.com/ropensci/biomartr/issues/31" class="external-link">#31</a> <a href="https://github.com/ropensci/biomartr/issues/61" class="external-link">#61</a>)</p></li>
<li><p>in all <code>get*()</code> functions, the <code>readr::write_tsv(path = )</code> was exchanged to <code>readr::write_tsv(file = )</code>, since the <code>readr</code> package version &gt; 1.4.0 is depreciating the <code>path</code> argument.</p></li>
<li><p><code>tbl_df()</code> was deprecated in dplyr 1.0.0. Please use <code><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">tibble::as_tibble()</a></code> instead. -&gt; adjusted <code><a href="../reference/organismBM.html">organismBM()</a></code> accordingly</p></li>
<li><p><code>custom_download()</code>, <code><a href="../reference/getGENOMEREPORT.html">getGENOMEREPORT()</a></code>, and other download functions now have specified <code>withr::local_options(timeout = max(30000000, getOption("timeout")))</code> which extends the default 60sec timeout to 30000000sec</p></li>
</ul></div>
<div class="section level4">
<h4 id="bug-fixes-1-0-2">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-0-2"></a></h4>
<ul><li><p>Fixing bug where genome availability check in <code><a href="../reference/getCollection.html">getCollection()</a></code> was only performed in <code>NCBI RefSeq</code> and not in other databases due to a constant used in <code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> rather than a variable (Many thanks to Takahiro Yamada for catching the bug) <a href="https://github.com/ropensci/biomartr/issues/53" class="external-link">#53</a></p></li>
<li><p>fixing an issue that caused the <code><a href="../reference/read_cds.html">read_cds()</a></code> function to fail in <code>data.table</code> mode (Many thanks to Clement Kent) <a href="https://github.com/ropensci/biomartr/issues/57" class="external-link">#57</a></p></li>
<li><p>fixing an <code>SSL</code> bug that was found on <code>Ubuntu 20.04</code> systems <a href="https://github.com/ropensci/biomartr/issues/66" class="external-link">#66</a> (Many thanks to Håkon Tjeldnes)</p></li>
<li><p>fixing global variable issue that caused <code>clean.retrieval()</code> to fail when no documentation file was in a <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> folder</p></li>
<li><p>The NCBI recently started adding <code>NA</code> values as FTP file paths in their <code>species summary files</code> for species without reference genomes. As a result <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> stopped working, because no FTP paths were found for some species. This issue was now fixed by adding the filter rule <code>!is.na(ftp_path)</code> into all <code>get*()</code> functions (Many thanks for making me aware of this issue Ashok Kumar Sharma <a href="https://github.com/ropensci/biomartr/issues/34" class="external-link">#34</a> and Dominik Merges <a href="https://github.com/ropensci/biomartr/issues/72" class="external-link">#72</a>)</p></li>
<li><p>Fixing an issue in <code>custom_download()</code> where the <code>method</code> argument was causing issues when downloading from <code>https</code> directed <code>ftp</code> sites (Many thanks to <a href="https://github.com/cmatKhan" class="external-link">@cmatKhan</a>) <a href="https://github.com/ropensci/biomartr/issues/76" class="external-link">#76</a></p></li>
<li><p>Fixing issue when trying to combine multiple summary-stats files where NA’s were present in the list item that was passed along for combination in <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> <a href="https://github.com/ropensci/biomartr/issues/73" class="external-link">#73</a> (Many thanks to Dominik Merges)</p></li>
<li><p>Fixing a bug in <code><a href="../reference/download.database.all.html">download.database.all()</a></code> where the lack of removing listed file <code>*-metadata.json</code> caused corruption of the download process (Many thanks to Jaruwatana Lotharukpong)</p></li>
</ul><p>biomartr 0.9.2 - minor changes to comply with CRAN policy regarding Internet access failure -&gt; Instead of using warnings or error messages, only gentle messages are allowed to be used</p>
</div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.9.0" id="biomartr-090">biomartr 0.9.0<small>2019-05-21</small><a class="anchor" aria-label="anchor" href="#biomartr-090"></a></h2>
<p><strong>Please be aware that as of April 2019, ENSEMBLGENOMES was retired (<a href="https://www.ensembl.info/2019/03/08/joint-rest-server-for-ensembl-and-ensembl-genomes-in-ensembl-96/" class="external-link">see details here</a>). Hence, all <code>biomartr</code> functions were updated and won’t support data retrieval from <code>ENSEMBLGENOMES</code> servers anymore.</strong></p>
<div class="section level4">
<h4 id="new-functions-0-9-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-9-0"></a></h4>
<ul><li>New function <code>clean.retrieval()</code> enables formatting and automatic unzipping of meta.retrieval output (find out more here: <a href="https://docs.ropensci.org/biomartr/articles/MetaGenome_Retrieval.html#un-zipping-downloaded-files" class="external-link uri">https://docs.ropensci.org/biomartr/articles/MetaGenome_Retrieval.html#un-zipping-downloaded-files</a>)</li>
<li>New function <code><a href="../reference/getGenomeSet.html">getGenomeSet()</a></code> allows users to easily retrieve genomes of multiple specified species. In addition, the genome summary statistics for all retrieved species will be stored as well to provide users with insights regarding the genome assembly quality of each species. This file can be used as Supplementary Information file in publications to facilitate reproducible research.</li>
<li>New function <code><a href="../reference/getProteomeSet.html">getProteomeSet()</a></code> allows users to easily retrieve proteomes of multiple specified species</li>
<li>New function <code><a href="../reference/getCDSSet.html">getCDSSet()</a></code> allows users to easily retrieve coding sequences of multiple specified species</li>
<li>New function <code><a href="../reference/getGFFSet.html">getGFFSet()</a></code> allows users to easily retrieve GFF annotation files of multiple specified species</li>
<li>New function <code><a href="../reference/getRNASet.html">getRNASet()</a></code> allows users to easily retrieve RNA sequences of multiple specified species</li>
<li>New function <code><a href="../reference/summary_genome.html">summary_genome()</a></code> allows users to retrieve summary statistics for a genome assembly file to assess the influence of genome assembly qualities when performing comparative genomics tasks</li>
<li>New function <code><a href="../reference/summary_cds.html">summary_cds()</a></code> allows users to retrieve summary statistics for a coding sequence (CDS) file. We noticed, that many CDS files stored in NCBI or ENSEMBL databases contain sequences that aren’t divisible by 3 (division into codons). This makes it difficult to divide CDS into codons for e.g. codon alignments or translation into protein sequences. In addition, some CDS files contain a significant amount of sequences that do not start with AUG (start codon). This function enables users to quantify how many of these sequences exist in a downloaded CDS file to process these files according to the analyses at hand.</li>
</ul></div>
<div class="section level4">
<h4 id="new-features-of-existing-functions-0-9-0">New Features of Existing Functions<a class="anchor" aria-label="anchor" href="#new-features-of-existing-functions-0-9-0"></a></h4>
<ul><li>the default value of argument <code>reference</code> in <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> changed from <code>reference = TRUE</code> to <code>reference = FALSE</code>. This way all genomes (reference AND non-reference) genomes will be downloaded by default. This is what users seem to prefer.</li>
<li>
<code><a href="../reference/getCollection.html">getCollection()</a></code> now also retrieves <code>GTF</code> files when <code>db = 'ensembl'</code>
</li>
<li>
<code><a href="../reference/getAssemblyStats.html">getAssemblyStats()</a></code> now also performs md5 checksum test</li>
<li>all md5 checksum tests now retrieve the new md5checkfile format from NCBI RefSeq and Genbank</li>
<li>
<code><a href="../reference/getGTF.html">getGTF()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID in addition to the scientific name in argument ‘organism’ to retrieve genome assemblies</li>
<li>
<code><a href="../reference/getGFF.html">getGFF()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID for ENSEMBL in addition to the scientific name in argument ‘organism’ to retrieve genome assemblies</li>
<li>
<code><a href="../reference/getMarts.html">getMarts()</a></code> will now throw an error when BioMart servers cannot be reached (<a href="https://github.com/ropensci/biomartr/issues/36" class="external-link">#36</a>)</li>
<li>
<code><a href="../reference/getGenome.html">getGenome()</a></code> now also stores the genome summary statistics (see <code>?summary_genome()</code>) for the retrieved species in the <code>documentation</code> folder to provide users with insights regarding the genome assembly quality</li>
<li>In all get*() functions the default for argument <code>reference</code> is now set from <code>reference = TRUE</code> to <code>reference = FALSE</code> (= new default)</li>
<li>all <code>get*()</code> functions now received a new argument <code>release</code> which allows users to retrieve specific release versions of genomes, proteomes, etc from <code>ENSEMBL</code> and <code>ENSEMBLGENOMES</code>
</li>
<li>all <code>get*()</code> functions received two new arguments <code>clean_retrieval</code> and <code>gunzip</code> which allows users to upzip the downloaded files directly in the <code>get*()</code> function call and rename the file for more convenient downstream analyses</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.8.0" id="biomartr-080">biomartr 0.8.0<small>2018-06-27</small><a class="anchor" aria-label="anchor" href="#biomartr-080"></a></h2>
<div class="section level4">
<h4 id="new-functions-0-8-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-8-0"></a></h4>
<ul><li>new function <code><a href="../reference/getCollection.html">getCollection()</a></code> for retrieval of a collection: the genome sequence, protein sequences, gff files, etc for a particular species</li>
</ul></div>
<div class="section level4">
<h4 id="new-functionality-of-existing-functions-0-8-0">New Functionality of Existing Functions<a class="anchor" aria-label="anchor" href="#new-functionality-of-existing-functions-0-8-0"></a></h4>
<ul><li><p><code><a href="../reference/getProteome.html">getProteome()</a></code> can now retrieve proteomes from the <a href="https://www.uniprot.org/" class="external-link">UniProt</a> database by specifying <code>getProteome(db = "uniprot")</code>.</p></li>
<li><p><code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> now prints out more useful interactive messages when searching for available organisms</p></li>
<li><p><code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> can now handle <code>taxids</code> and <code>assembly_accession ids</code> in addition to the scientific name when specifying argument <code>organism</code></p></li>
<li><p><code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> can now check for organism availability in the UniProt database</p></li>
<li><p><code><a href="../reference/getGenome.html">getGenome()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID in addition to the scientific name in argument ‘organism’ to retrieve genome assemblies</p></li>
<li><p><code><a href="../reference/getProteome.html">getProteome()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID in addition to the scientific name in argument ‘organism’ to retrieve proteomes</p></li>
<li><p><code><a href="../reference/getCDS.html">getCDS()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID in addition to the scientific name in argument ‘organism’ to retrieve CDS</p></li>
<li><p><code><a href="../reference/getRNA.html">getRNA()</a></code>: users can now specify the NCBI Taxonomy ID or Accession ID in addition to the scientific name in argument ‘organism’ to retrieve RNAs</p></li>
<li><p><code><a href="../reference/is.genome.available.html">is.genome.available()</a></code>: argument order was changed from is.genome.available(organism, details, db) to is.genome.available(db, organism, details) to be logically more consistent with all <code>get*()</code> functions</p></li>
<li><p><code>meta.retrieval</code> receives a new argument <code>restart_at_last</code> to indicate whether or not the download process when re-running the <code>meta.retrieval</code> function shall pick up at the last species or whether it should crawl through all existing files to check the md5checksum</p></li>
<li><p><code>meta.retrieval</code> now generates an csv overview file in the <code>doc</code> folder which stores genome version, date, origin, etc information for all downloaded organisms and can be directly used as Supplementary Data file in publications to increase computational and biological reproducibility of the genomics study</p></li>
<li><p><code><a href="../reference/download.database.all.html">download.database.all()</a></code> can now skip already downloaded files and internally removes corrupted files with non-matching md5checksum. Re-downloading of currupted files and be performed by simply re-running the <code><a href="../reference/download.database.all.html">download.database.all()</a></code> command</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.7.0" id="biomartr-070">biomartr 0.7.0<small>2018-01-03</small><a class="anchor" aria-label="anchor" href="#biomartr-070"></a></h2>
<div class="section level4">
<h4 id="function-changes-0-7-0">Function changes<a class="anchor" aria-label="anchor" href="#function-changes-0-7-0"></a></h4>
<ul><li><p>the function <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> will now pick up the download at the organism where it left off and will report which species have already been retrieved</p></li>
<li><p>all <code>get*()</code> functions and the <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> function receive a new argument <code>reference</code> which allows users to retrieve not-reference or not-representative genome versions when downloading from NCBI RefSeq or NCBI Genbank</p></li>
<li><p>the argument order in <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> changed from <code>meta.retrieval(kingdom, group, db, ...)</code> to <code>meta.retrieval(db,kingdom, group, ...)</code> to make the argument order more consistent with the <code>get*()</code> functions</p></li>
<li><p>the argument order in <code><a href="../reference/getGroups.html">getGroups()</a></code> changed from <code>getGroups(kingdom, db)</code> to <code>getGroups(db, kingdom)</code> to make the argument order more consistent with the <code>get*()</code> and <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> functions</p></li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-7-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-7-0"></a></h4>
<ul><li>new internal functions <code>existingOrganisms()</code> and <code>existingOrganisms_ensembl()</code> which check the organisms that have already been downloaded</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.2" id="biomartr-052">biomartr 0.5.2<small>2017-09-20</small><a class="anchor" aria-label="anchor" href="#biomartr-052"></a></h2>
<div class="section level4">
<h4 id="bug-fixes-0-5-2">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-5-2"></a></h4>
<ul><li>fixing bug (<a href="https://github.com/ropensci/biomartr/issues/6" class="external-link uri">https://github.com/ropensci/biomartr/issues/6</a>) that caused incorrect filtering condition when more than one entry for an organism is present in the assemblysummary.txt file at NCBI (Thanks to <a href="https://github.com/kalmeshv" class="external-link">@kalmeshv</a>)</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.5.1" id="biomartr-051">biomartr 0.5.1<small>2017-05-28</small><a class="anchor" aria-label="anchor" href="#biomartr-051"></a></h2>
<div class="section level4">
<h4 id="bug-fixes-0-5-1">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-5-1"></a></h4>
<ul><li><p>fixing a bug in <code>exists.ftp.file()</code> and <code>getENSEMBLGENOMES.Seq()</code> that caused bacterial genome, proteome, etc retrieval to fail due to the wrong construction of a query ftp request <a href="https://github.com/ropensci/biomartr/issues/7" class="external-link uri">https://github.com/ropensci/biomartr/issues/7</a> (Many thanks to <a href="https://github.com/dbsseven" class="external-link">@dbsseven</a>)</p></li>
<li><p>fix a major bug in which organisms having no representative genome would generate NULL paths that subsequently crashed the <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> function when it tried to print out the result paths.</p></li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-5-1">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-5-1"></a></h4>
<ul><li><p>new function <code><a href="../reference/getRepeatMasker.html">getRepeatMasker()</a></code> for retrieval of Repeat Masker output files</p></li>
<li><p>new function <code><a href="../reference/getGTF.html">getGTF()</a></code> for genome annotation retrieval from <code>ensembl</code> and <code>ensemblgenomes</code> in <code>gtf</code> format (Thanks for suggesting it Ge Tan)</p></li>
<li><p>new function <code><a href="../reference/getRNA.html">getRNA()</a></code> to perform RNA Sequence Retrieval from NCBI and ENSEMBL databases (Thanks for suggesting it <a href="https://github.com/carlo-berg" class="external-link">@carlo-berg</a>)</p></li>
<li><p>new function <code><a href="../reference/read_rna.html">read_rna()</a></code> for importing Repeat Masker output files downloaded with <code><a href="../reference/getRepeatMasker.html">getRepeatMasker()</a></code></p></li>
<li><p>new function <code><a href="../reference/read_rm.html">read_rm()</a></code> for importing RNA downloaded with <code><a href="../reference/getRNA.html">getRNA()</a></code> as Biostrings or data.table object</p></li>
<li><p>new helper function <code>custom_download()</code> that aims to make the download process more robust and stable -&gt; In detail, the download process is now adapting to the operating system, e.g. using either <code>curl</code> (macOS), <code>wget</code> (Linux), or <code>wininet</code> (Windows)</p></li>
</ul></div>
<div class="section level4">
<h4 id="function-changes-0-5-1">Function changes<a class="anchor" aria-label="anchor" href="#function-changes-0-5-1"></a></h4>
<ul><li><p>function name <code><a href="../reference/listDatabases.html">listDatabases()</a></code> has been renamed <code><a href="../reference/listDatabases.html">listNCBIDatabases()</a></code>. In <code>biomartr</code> version 0.6.0 the function name <code><a href="../reference/listDatabases.html">listDatabases()</a></code> will be depreciated</p></li>
<li><p><code>meta.retieval()</code> and <code>meta.retieval.all()</code> now allow the bulk retrieval of GTF files for <code>type = 'ensembl'</code> and <code>type = 'esnemblgenomes'</code> via <code>type = "gtf"</code>. See <code><a href="../reference/getGTF.html">getGTF()</a></code> for more details.</p></li>
<li><p><code>meta.retieval()</code> and <code>meta.retieval.all()</code> now allow the bulk retrieval of RNA files via <code>type = "rna"</code>. See <code><a href="../reference/getRNA.html">getRNA()</a></code> for more details.</p></li>
<li><p><code>meta.retieval()</code> and <code>meta.retieval.all()</code> now allow the bulk retrieval of Repeat Masker output files via <code>type = "rm"</code>. See <code><a href="../reference/getRepeatMasker.html">getRepeatMasker()</a></code> for more details.</p></li>
<li><p>all <code>get*()</code> retrieval functions now skip the download of a particular file if it already exists in the specified file path</p></li>
<li><p><code><a href="../reference/download.database.html">download.database()</a></code> and <code><a href="../reference/download.database.all.html">download.database.all()</a></code> now internally perform md5 check sum checks to make sure that the file download was successful</p></li>
<li><p><code><a href="../reference/download.database.html">download.database()</a></code> and <code><a href="../reference/download.database.all.html">download.database.all()</a></code> now return the file paths of the downloaded file so that it is easier to use these functions when constructing pipelines, e.g. <code>download.database() %&gt;% ...</code> or <code>download.database.all() %&gt;% ...</code>.</p></li>
<li><p><code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> and <code><a href="../reference/meta.retrieval.all.html">meta.retrieval.all()</a></code> now return the file paths of the downloaded file so that it is easier to use these functions when constructing pipelines, e.g. <code>meta.retrieval() %&gt;% ...</code> or <code>meta.retrieval() %&gt;% ...</code>.</p></li>
<li><p><code><a href="../reference/getGenome.html">getGenome()</a></code>, <code><a href="../reference/getProteome.html">getProteome()</a></code>, <code><a href="../reference/getCDS.html">getCDS()</a></code>, <code><a href="../reference/getRNA.html">getRNA()</a></code>, <code><a href="../reference/getGFF.html">getGFF()</a></code>, and <code><a href="../reference/getAssemblyStats.html">getAssemblyStats()</a></code> now internally perform md5 checksum tests to make sure that files are retrieved intact.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.4.0" id="biomartr-040">biomartr 0.4.0<small>2017-03-14</small><a class="anchor" aria-label="anchor" href="#biomartr-040"></a></h2>
<div class="section level4">
<h4 id="bug-fixes-0-4-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-4-0"></a></h4>
<ul><li>fixing a major bug <a href="https://github.com/ropensci/biomartr/issues/6" class="external-link uri">https://github.com/ropensci/biomartr/issues/6</a> that caused that in all <code>get*()</code> (genome, proteome, gff, etc.) and <code>meta.retrieval*()</code> functions the meta retrieval process errored and terminated whenever NCBI or ENSEMBL didn’t store all types of sequences for a particular organism: genome, proteome, cds, etc. This has been fixed now and function calls such as <code>meta.retrieval(kingdom = "bacteria", db = "genbank", type = "proteome")</code> should work properly now (Thanks to <a href="https://github.com/ARamesh123" class="external-link">@ARamesh123</a> for making me aware if this bug). Hence, this bug affected all attempts to download all proteome sequences e.g. for bacteria and viruses, because NCBI does not store genome AND proteome information for all bacterial or viral species.</li>
</ul></div>
<div class="section level4">
<h4 id="new-functions-0-4-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-4-0"></a></h4>
<ul><li><p>new function <code><a href="../reference/getAssemblyStats.html">getAssemblyStats()</a></code> allows users to retrieve the genome assembly stats file from NCBI RefSeq or Genbank, e.g. <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.36_GRCh38.p10/GCF_000001405.36_GRCh38.p10_assembly_stats.txt" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.36_GRCh38.p10/GCF_000001405.36_GRCh38.p10_assembly_stats.txt</a></p></li>
<li><p>new function <code><a href="../reference/read_assemblystats.html">read_assemblystats()</a></code> allows to import the genome assembly stats file from NCBI RefSeq or Genbank that was retrieved using the <code><a href="../reference/getAssemblyStats.html">getAssemblyStats()</a></code> function</p></li>
</ul></div>
<div class="section level4">
<h4 id="function-changes-0-4-0">Function changes<a class="anchor" aria-label="anchor" href="#function-changes-0-4-0"></a></h4>
<ul><li><p><code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> and <code><a href="../reference/meta.retrieval.all.html">meta.retrieval.all()</a></code> can now also download genome assembly stats for all selected species</p></li>
<li><p><code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> receives a new argument <code>group</code> that allows users to retrieve species belonging to a subgroup instead of the entire kingdom. Available groups can be retrieved with <code><a href="../reference/getGroups.html">getGroups()</a></code>.</p></li>
<li><p>functions <code>getSubgroups()</code> and <code>listSubgroups()</code> have been removed and their initial functionality has been merged and integrated into <code><a href="../reference/getGroups.html">getGroups()</a></code> and <code><a href="../reference/listGroups.html">listGroups()</a></code></p></li>
<li><p><code><a href="../reference/listGroups.html">listGroups()</a></code> receives a new argument <code>details</code> that allows users to retrieve the organism names that belong to the corresponding subgroups</p></li>
<li><p><code><a href="../reference/getGroups.html">getGroups()</a></code> is now based on <code><a href="../reference/listGroups.html">listGroups()</a></code></p></li>
<li><p>internal function <code>getGENOMESREPORT()</code> is now exported and available to the user</p></li>
<li><p>all <code>organism*()</code> functions now also support Ensembl Plants, Ensembl Metazoa, Ensembl Protist, and Ensembl Fungi (Thanks for pointing out <a href="https://github.com/AlexGa" class="external-link">Alex Gabel</a>)</p></li>
<li><p><code><a href="../reference/getMarts.html">getMarts()</a></code> and <code><a href="../reference/getDatasets.html">getDatasets()</a></code> now also support Ensembl Plants, Ensembl Metazoa, Ensembl Protist, and Ensembl Fungi (Thanks for pointing out <a href="https://github.com/AlexGa" class="external-link">Alex Gabel</a>)</p></li>
</ul></div>
<div class="section level4">
<h4 id="vignette-updates-0-4-0">Vignette updates<a class="anchor" aria-label="anchor" href="#vignette-updates-0-4-0"></a></h4>
<ul><li>Vignette <code>Meta-Genome Retrieval</code> has more examples how to download genomes of species that belong to the same subgroup</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.3.0" id="biomartr-030">biomartr 0.3.0<small>2017-02-09</small><a class="anchor" aria-label="anchor" href="#biomartr-030"></a></h2>
<div class="section level4">
<h4 id="bug-fixes-0-3-0">Bug fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-3-0"></a></h4>
<ul><li>Fixing a bug <a href="https://github.com/ropensci/biomartr/issues/2" class="external-link uri">https://github.com/ropensci/biomartr/issues/2</a> based on the <a href="https://github.com/tidyverse/readr" class="external-link">readr package</a> that affected the <code><a href="../reference/getSummaryFile.html">getSummaryFile()</a></code>, <code><a href="../reference/getKingdomAssemblySummary.html">getKingdomAssemblySummary()</a></code>, <code><a href="../reference/getMetaGenomeSummary.html">getMetaGenomeSummary()</a></code>, <code><a href="../reference/getENSEMBL.Seq.html">getENSEMBL.Seq()</a></code> and <code>getENSEMBLGENOMES.Seq()</code> functions causing quoted lines in the <code>assembly_summary.txt</code> to be omitted when reading these files. This artefact caused that e.g. instead of information of 80,000 Bacteria genomes only 40,000 (which non-quotations) were read (Thanks to <a href="https://github.com/alartin" class="external-link">Xin Wu</a>).</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.2.1" id="biomartr-021">biomartr 0.2.1<small>2016-12-15</small><a class="anchor" aria-label="anchor" href="#biomartr-021"></a></h2>
<p>In this version of <code>biomartr</code> the <code>organism*()</code> functions were adapted to the new <a href="https://www.ensembl.info/2016/12/08/ensembl-87-has-been-released/" class="external-link">ENSEMBL 87 release</a> in which organism name specification in the Biomart description column <a href="https://github.com/ropensci/biomartr/issues/1" class="external-link">was changed</a> from a scientific name convention to a mix of common name and scientific name convention.</p>
<ul><li><p>all <code>organism*()</code> functions have been adapted to the new ENSEMBL 87 release organism name notation that is used in the Biomart description</p></li>
<li><p>fixing error handling bug that caused commands such as <code>download.database(db = "nr.27.tar.gz")</code> to not execute properly</p></li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.2.0" id="biomartr-020">biomartr 0.2.0<small>2016-11-22</small><a class="anchor" aria-label="anchor" href="#biomartr-020"></a></h2>
<p>In this version, <code>biomartr</code> was extended to now retrieve genome, proteome, CDS, GFF and meta-genome data also from <a href="https://www.ensembl.org/index.html" class="external-link">ENSEMBL</a> and <a href="https://ensemblgenomes.org/" class="external-link">ENSEMLGENOMES</a>. Furthermore, all NCBI retrieval functions were updated to the new server folder structure standards of NCBI.</p>
<div class="section level4">
<h4 id="new-functions-0-2-0">New Functions<a class="anchor" aria-label="anchor" href="#new-functions-0-2-0"></a></h4>
<ul><li><p>new meta-retrieval function <code><a href="../reference/meta.retrieval.all.html">meta.retrieval.all()</a></code> allows users to download all individual genomes of all kingdoms of life with one command</p></li>
<li><p>new metagenome retrieval function <code><a href="../reference/getMetaGenomes.html">getMetaGenomes()</a></code> allows users to retrieve metagenome projects from NCBI Genbank</p></li>
<li><p>new metagenome retrieval function <code><a href="../reference/getMetaGenomeAnnotations.html">getMetaGenomeAnnotations()</a></code> allows users to retrieve annotation files for genomes belonging to a metagenome project stored at NCBI Genbank</p></li>
<li><p>new retrieval function <code><a href="../reference/getGFF.html">getGFF()</a></code> allows users to retrieve annotation (*.gff) files for specific genomes from NCBI and ENSEMBL databases</p></li>
<li><p>new import function <code><a href="../reference/read_gff.html">read_gff()</a></code> allowing users to import GFF files downloaded with <code><a href="../reference/getGFF.html">getGFF()</a></code></p></li>
<li><p>new internal functions to check for availability of ENSEMBL or ENSEMBLGENOMES databases</p></li>
<li><p>new database retrieval function <code><a href="../reference/download.database.all.html">download.database.all()</a></code> allows users to download entire NCBI databases with one command</p></li>
<li><p>new function <code><a href="../reference/listMetaGenomes.html">listMetaGenomes()</a></code> allowing users to list available metagenomes on NCBI Genbank</p></li>
<li><p>new external helper function <code><a href="../reference/getSummaryFile.html">getSummaryFile()</a></code> to retrieve the assembly_summary.txt file from NCBI</p></li>
<li><p>new external helper function <code><a href="../reference/getKingdomAssemblySummary.html">getKingdomAssemblySummary()</a></code> to retrieve the assembly_summary.txt files from NCBI for all kingdoms and combine them into one big data.frame</p></li>
<li><p>new function <code><a href="../reference/listKingdoms.html">listKingdoms()</a></code> allows users to list the number of available species per kingdom of life</p></li>
<li><p>new function <code><a href="../reference/listGroups.html">listGroups()</a></code> allows users to list the number of available species per group</p></li>
<li><p>new function <code>listSubgroups()</code> allows users to list the number of available species per subgroup</p></li>
<li><p>new function <code><a href="../reference/getGroups.html">getGroups()</a></code> allows users to retrieve available groups for a kingdom of life</p></li>
<li><p>new function <code>getSubgroups()</code> allows users to retrieve available subgroups for a kingdom of life</p></li>
<li><p>new external helper function <code><a href="../reference/getMetaGenomeSummary.html">getMetaGenomeSummary()</a></code> to retrieve the assembly_summary.txt files from NCBI genbank metagenomes</p></li>
<li><p>new internal helper function <code><a href="../reference/getENSEMBL.Seq.html">getENSEMBL.Seq()</a></code> acting as main interface function to communicate with the ENSEMBL database API for sequence retrieval</p></li>
<li><p>new internal helper function <code>getENSEMBLGENOMES.Seq()</code> acting as main interface function to communicate with the ENSEMBL database API for sequence retrieval</p></li>
<li><p>new internal helper function <code>getENSEMBL.Annotation()</code> acting as main interface function to communicate with the ENSEMBL database API for GFF retrieval</p></li>
<li><p>new internal helper function <code>getENSEMBLGENOMES.Annotation()</code> acting as main interface function to communicate with the ENSEMBL database API for GFF retrieval</p></li>
<li><p>new internal helper function <code>get.ensemblgenome.info()</code> to retrieve general organism information from ENSEMBLGENOMES</p></li>
<li><p>new internal helper function <code><a href="../reference/get.ensembl.info.html">get.ensembl.info()</a></code> to retrieve general organism information from ENSEMBL</p></li>
<li><p>new internal helper function <code><a href="../reference/getGENOMEREPORT.html">getGENOMEREPORT()</a></code> to retrieve the genome reports file from <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/GENOME_REPORTS/overview.txt" class="uri">ftp://ftp.ncbi.nlm.nih.gov/genomes/GENOME_REPORTS/overview.txt</a></p></li>
<li><p>new internal helper function <code>connected.to.internet()</code> enabling internet connection check</p></li>
</ul></div>
<div class="section level4">
<h4 id="function-changes-0-2-0">Function changes<a class="anchor" aria-label="anchor" href="#function-changes-0-2-0"></a></h4>
<ul><li><p>functions <code><a href="../reference/getGenome.html">getGenome()</a></code>, <code><a href="../reference/getProteome.html">getProteome()</a></code>, and <code><a href="../reference/getCDS.html">getCDS()</a></code> now can also in addition to NCBI retrieve genomes, proteomes or CDS from <a href="https://www.ensembl.org/index.html" class="external-link">ENSEMBL</a> and <a href="https://ensemblgenomes.org/" class="external-link">ENSEMLGENOMES</a></p></li>
<li><p>the functions <code><a href="../reference/getGenome.html">getGenome()</a></code>, <code><a href="../reference/getProteome.html">getProteome()</a></code>, and <code><a href="../reference/getCDS.html">getCDS()</a></code> were completely re-written and now use the assembly_summary.txt files provided by NCBI to retrieve the download path to the corresponding genome. Furthermore, these functions now lost the <code>kingdom</code> argument. Users now only need to specify the organism name and not the kingdom anymore. Furthermore, all <code>get*</code> functions now return the path to the downloaded genome so that this path can be used as input to all <code>read_*</code> functions.</p></li>
<li><p><code>download_databases()</code> has been renamed to <code>download.databases()</code> to be more consistent with other function notation</p></li>
<li><p>the argument <code>db_format</code> was removed from <code><a href="../reference/listDatabases.html">listDatabases()</a></code> and <code><a href="../reference/download.database.html">download.database()</a></code> because it was misleading</p></li>
<li><p>the command <code>listDatabases("all")</code> now returns all available NCBI databases that can be retrieved with <code><a href="../reference/download.database.html">download.database()</a></code></p></li>
<li><p><code><a href="../reference/download.database.html">download.database()</a></code> now internally checks if input database specified by the user is actually available on NCBI servers</p></li>
<li><p>the documentary file generated by <code><a href="../reference/getGenome.html">getGenome()</a></code>, <code><a href="../reference/getProteome.html">getProteome()</a></code>, and <code><a href="../reference/getCDS.html">getCDS()</a></code> is now extended to store more details about the downloaded genome</p></li>
<li><p>argument <code>database</code> in <code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> and <code><a href="../reference/listGenomes.html">listGenomes()</a></code> has been renamed to <code>db</code> to be consistent with all other sequence retrieval functions</p></li>
<li><p><code><a href="../reference/is.genome.available.html">is.genome.available()</a></code> now also checks availability of organisms in ENSEMBL. See <code>db = "ensembl"</code></p></li>
<li><p>the argument <code>db_name</code> in <code><a href="../reference/listDatabases.html">listDatabases()</a></code> has been renamed <code>db</code> to be more consistent with the notation in other functions</p></li>
<li><p>the argument <code>name</code> in <code><a href="../reference/download.database.html">download.database()</a></code> has been renamed <code>db</code> to be more consistent with the notation in other functions</p></li>
<li><p><code><a href="../reference/getKingdoms.html">getKingdoms()</a></code> now retrieves also kingdom information for ENSEMBL and ENSEMBLGENOMES</p></li>
<li><p><code><a href="../reference/getKingdoms.html">getKingdoms()</a></code> received new argument <code>db</code> to specify from which database (e.g. <code>refseq</code>, <code>genbank</code>, <code>ensembl</code> or <code>ensemblgenomes</code>) kingdom information shall be retrieved</p></li>
<li><p><code>getKingdoms(db = "refseq")</code> received one more member: <code>"viral"</code>, allowing the genome retrieval of all viruses</p></li>
<li><p>argument <code>out.folder</code> in <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> has been renamed to <code>path</code> to be more consistent with other retrieval functions</p></li>
<li><p>all <code>read_*</code> functions now received a new argument <code>obj.type</code> allowing users to choose between storing input genomes as Biostrings object or data.table object</p></li>
<li><p>all <code>read_*</code> functions now have <code>format = "fasta"</code> as default</p></li>
<li><p>the <code>kingdom</code> argument in the <code><a href="../reference/listGenomes.html">listGenomes()</a></code> function was renamed to <code>type</code>, now allowing users to specify not only specify kingdoms, but also groups and subgroups. Use: <code>listGenomes(type = "kingdom")</code> or <code>listGenomes(type = "group")</code> or <code>listGenomes(type = "subgroup")</code></p></li>
<li><p>the <code><a href="../reference/listGenomes.html">listGenomes()</a></code> function receives a new argument <code>subset</code> to specify a subset of the selected <code>type</code> argument. E.g. <code>subset = "Eukaryota"</code> when specifying <code>type = "kingdom"</code></p></li>
</ul></div>
<div class="section level4">
<h4 id="vignette-updates-0-2-0">Vignette updates<a class="anchor" aria-label="anchor" href="#vignette-updates-0-2-0"></a></h4>
<ul><li>new Vignette <code>Meta-Genome Retrieval</code>
</li>
<li>Update examples and extend <code>Introduction</code> Vignette</li>
<li>Update examples and extend <code>Database Retrieval</code> Vignette</li>
<li>Update examples and extend <code>Sequence Retrieval</code> Vignette</li>
<li>Update examples and extend <code>Functional Annotation</code> Vignette</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.1.0" id="biomartr-010">biomartr 0.1.0<small>2016-08-07</small><a class="anchor" aria-label="anchor" href="#biomartr-010"></a></h2>
<ul><li><p>fixing a parsing error of the file <code>ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/assembly_summary.txt</code> The problem was that comment lines were introduced and columns couldn’t be parsed correctly anymore. This caused that genomes, proteomes, and CDS files could not be downloaded properly. This has been fixed now.</p></li>
<li><p>genomes, proteome, and CDS as well as meta-genomes can now be retrieved from RefSeq and Genbank (not only RefSeq); only <code><a href="../reference/getCDS.html">getCDS()</a></code> does not have genebank access, becasue genbank does not provide CDS sequences</p></li>
<li><p>adding new function <code><a href="../reference/meta.retrieval.html">meta.retrieval()</a></code> to mass retrieve genomes for entire kingdoms of life</p></li>
<li><p>fixed a major bug in <code><a href="../reference/organismBM.html">organismBM()</a></code> causing the function to fail. The failure of this function affected all downstream <code>organism*()</code> functions. Bug is now fixed and everything works properly</p></li>
<li><p>updated Vignettes</p></li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.3" id="biomartr-003">biomartr 0.0.3<small>2016-03-02</small><a class="anchor" aria-label="anchor" href="#biomartr-003"></a></h2>
<ul><li><p>updating unit tests for new API</p></li>
<li><p>fixing API problems that caused all BioMart related functions to fail</p></li>
<li><p>fixing retrieval problems in <code><a href="../reference/getCDS.html">getCDS()</a></code>, <code><a href="../reference/getProteome.html">getProteome()</a></code>, and <code><a href="../reference/getGenome.html">getGenome()</a></code></p></li>
<li><p>the <code><a href="../reference/listDatabases.html">listDatabases()</a></code> function now has a new option <code>db_name = "all"</code> allowing users to list all available databases stored on NCBI</p></li>
</ul><div class="section level4">
<h4 id="vignettes-0-0-3">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-0-3"></a></h4>
<ul><li>adding new vignette: Database Retrieval</li>
<li>update the vignettes: Phylotranscriptomics, Sequence Retrieval, and Functional Annotation</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.2" id="biomartr-002">biomartr 0.0.2<small>2015-08-07</small><a class="anchor" aria-label="anchor" href="#biomartr-002"></a></h2>
<div class="section level4">
<h4 id="vignettes-0-0-2">Vignettes<a class="anchor" aria-label="anchor" href="#vignettes-0-0-2"></a></h4>
<ul><li>adding vignettes: Introduction, Functional Annotation, Phylotranscriptomics, and Sequence Retrieval</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.0.1" id="biomartr-001">biomartr 0.0.1<small>2015-07-22</small><a class="anchor" aria-label="anchor" href="#biomartr-001"></a></h2>
<p>Release Version</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

