<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Ensembl BioMart Examples â€¢ biomartr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Ensembl BioMart Examples">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Ensembl BioMart Examples</h1>
            
            <h4 data-toc-skip class="date">2025-07-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/vignettes/BioMart_Examples.Rmd" class="external-link"><code>vignettes/BioMart_Examples.Rmd</code></a></small>
      <div class="hidden name"><code>BioMart_Examples.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="use-case-1-functional-annotation-of-genes-sharing-a-common-evolutionary-history">Use Case #1: Functional Annotation of Genes Sharing a Common
Evolutionary History<a class="anchor" aria-label="anchor" href="#use-case-1-functional-annotation-of-genes-sharing-a-common-evolutionary-history"></a>
</h2>
<p>Evolutionary Transcriptomics aims to predict stages or periods of
evolutionary conservation in biological processes on the transcriptome
level. However, finding genes sharing a <a href="https://drostlab.github.io/myTAI/articles/Enrichment.html" class="external-link">common
evolutionary history</a> could reveal how the the biological process
might have evolved in the first place.</p>
<p>In this <code>Use Case</code> we will combine functional and
biological annotation obtained with <code>biomartr</code> with enriched
genes obtained with <a href="https://drostlab.github.io/myTAI/articles/Enrichment.html" class="external-link">PlotEnrichment()</a>.</p>
<div class="section level3">
<h3 id="step-1">Step 1<a class="anchor" aria-label="anchor" href="#step-1"></a>
</h3>
<p>For the following example we will use the dataset an enrichment
analyses found in <a href="https://drostlab.github.io/myTAI/articles/Enrichment.html" class="external-link">PlotEnrichment()</a>.</p>
<p>Install and load the <a href="https://github.com/drostlab/myTAI" class="external-link">myTAI</a> package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install myTAI</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"myTAI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load myTAI</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">myTAI</span><span class="op">)</span></span></code></pre></div>
<p>Download the <code>Phylostratigraphic Map</code> of <em>D.
rerio</em>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the Phylostratigraphic Map of Danio rerio</span></span>
<span><span class="co"># from Sestak and Domazet-Loso, 2015</span></span></code></pre></div>
<p>The dataset comes from <code>Supplementary file 3</code> of this
publication: <a href="https://academic.oup.com/mbe/article/32/2/299/1058654#77837069" class="external-link uri">https://academic.oup.com/mbe/article/32/2/299/1058654#77837069</a></p>
<p>After downloading <code>Supplementary file 3</code>, you will find
the file <code>TableS3-2.xlsx</code> which can be used in the following
<code>biomartr</code> functions.</p>
<p>Read the <code>*.xlsx</code> file storing the
<code>Phylostratigraphic Map</code> of <em>D. rerio</em> and format it
for the use with <code>myTAI</code>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install the readxl package</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"readxl"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># load package readxl</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org" class="external-link">readxl</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># read the excel file</span></span>
<span><span class="va">DrerioPhyloMap.MBEa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html" class="external-link">read_excel</a></span><span class="op">(</span><span class="st">"TableS3-2.xlsx"</span>, sheet <span class="op">=</span> <span class="fl">1</span>, skip <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># format Phylostratigraphic Map for use with myTAI</span></span>
<span><span class="va">Drerio.PhyloMap</span> <span class="op">&lt;-</span> <span class="va">DrerioPhyloMap.MBEa</span><span class="op">[</span> , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># have a look at the final format</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">Drerio.PhyloMap</span><span class="op">)</span></span></code></pre></div>
<pre><code>  Phylostrata            ZFIN_ID
1           1 ZDB-GENE-000208-13
2           1 ZDB-GENE-000208-17
3           1 ZDB-GENE-000208-18
4           1 ZDB-GENE-000208-23
5           1  ZDB-GENE-000209-3
6           1  ZDB-GENE-000209-4</code></pre>
<p>Now, <code>Drerio.PhyloMap</code> stores the
<code>Phylostratigraphic Map</code> of <em>D. rerio</em> which is used
as background set to perform enrichment analyses with
<code>PlotEnrichment()</code> from <code>myTAI</code>.</p>
</div>
<div class="section level3">
<h3 id="enrichment-analyses">Enrichment Analyses<a class="anchor" aria-label="anchor" href="#enrichment-analyses"></a>
</h3>
<p>Now, the <code>PlotEnrichment()</code> function visualizes the over-
and underrepresented <code>Phylostrata</code> of brain specific genes
when compared with the total number of genes stored in the
<code>Phylostratigraphic Map</code> of <em>D. rerio</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readxl.tidyverse.org" class="external-link">readxl</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># read expression data (organ specific genes) from Sestak and Domazet-Loso, 2015</span></span>
<span><span class="va">Drerio.OrganSpecificExpression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://readxl.tidyverse.org/reference/read_excel.html" class="external-link">read_excel</a></span><span class="op">(</span><span class="st">"TableS3-2.xlsx"</span>, sheet <span class="op">=</span> <span class="fl">2</span>, skip <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select only brain specific genes</span></span>
<span><span class="va">Drerio.Brain.Genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="va">Drerio.OrganSpecificExpression</span><span class="op">[</span> , <span class="st">"brain"</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># visualize enriched Phylostrata of genes annotated as brain specific</span></span>
<span><span class="fu">PlotEnrichment</span><span class="op">(</span><span class="va">Drerio.PhyloMap</span>,</span>
<span>               test.set     <span class="op">=</span> <span class="va">Drerio.Brain.Genes</span>,</span>
<span>               measure      <span class="op">=</span> <span class="st">"foldchange"</span>,</span>
<span>               use.only.map <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>               legendName   <span class="op">=</span> <span class="st">"PS"</span><span class="op">)</span></span></code></pre></div>
<p>Users will observe that for example brain genes deriving from PS5 are
significantly enriched.</p>
<p>Now we can select all brain genes originating in PS5 using the
<code>SelectGeneSet()</code> function from <code>myTAI</code>. Please
notice that <code>SelectGeneSet()</code> can only be used with
phylostratigraphic maps only (<code>use.map.only = TRUE</code> argument)
since myTAI version &gt; 0.3.0.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">BrainGenes</span> <span class="op">&lt;-</span> <span class="fu">SelectGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">Drerio.PhyloMap</span>,</span>
<span>                            gene.set      <span class="op">=</span> <span class="va">Drerio.Brain.Genes</span>,</span>
<span>                            use.only.map  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select only brain genes originating in PS5</span></span>
<span><span class="va">BrainGenes.PS5</span> <span class="op">&lt;-</span> <span class="va">BrainGenes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">BrainGenes</span><span class="op">[</span> , <span class="st">"Phylostrata"</span><span class="op">]</span> <span class="op">==</span> <span class="fl">5</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co"># look at the results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">BrainGenes.PS5</span><span class="op">)</span></span></code></pre></div>
<pre><code>      Phylostrata           ZFIN_ID
14851           5 ZDB-GENE-000210-6
14852           5 ZDB-GENE-000210-7
14853           5 ZDB-GENE-000328-4
14856           5 ZDB-GENE-000411-1
14857           5 ZDB-GENE-000427-4
14860           5 ZDB-GENE-000526-1</code></pre>
<p>Now users can perform the <code><a href="../reference/biomart.html">biomart()</a></code> function to obtain
the functional annotation of brain genes originating in PS5.</p>
<p>For this purpose, first we need to find the filter name of the
corresponding gene ids such as <code>ZDB-GENE-000210-6</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find filter for zfin.org ids</span></span>
<span><span class="fu"><a href="../reference/organismFilters.html">organismFilters</a></span><span class="op">(</span><span class="st">"Danio rerio"</span>, topic <span class="op">=</span> <span class="st">"zfin_id"</span><span class="op">)</span></span></code></pre></div>
<pre><code>                            name                           description             dataset
52                  with_zfin_id                       with ZFIN ID(s) drerio_gene_ensembl
53  with_zfin_id_transcript_name          with ZFIN transcript name(s) drerio_gene_ensembl
103                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-060825-136] drerio_gene_ensembl
274                 with_zfin_id                       with ZFIN ID(s)    drerio_gene_vega
286                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-121214-212]    drerio_gene_vega
366                 with_zfin_id                       with ZFIN ID(s) drerio_gene_ensembl
367 with_zfin_id_transcript_name          with ZFIN transcript name(s) drerio_gene_ensembl
417                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-060825-136] drerio_gene_ensembl
588                 with_zfin_id                       with ZFIN ID(s)    drerio_gene_vega
600                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-121214-212]    drerio_gene_vega
680                 with_zfin_id                       with ZFIN ID(s) drerio_gene_ensembl
681 with_zfin_id_transcript_name          with ZFIN transcript name(s) drerio_gene_ensembl
731                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-060825-136] drerio_gene_ensembl
902                 with_zfin_id                       with ZFIN ID(s)    drerio_gene_vega
914                      zfin_id ZFIN ID(s) [e.g. ZDB-GENE-121214-212]    drerio_gene_vega
                    mart
52  ENSEMBL_MART_ENSEMBL
53  ENSEMBL_MART_ENSEMBL
103 ENSEMBL_MART_ENSEMBL
274 ENSEMBL_MART_ENSEMBL
286 ENSEMBL_MART_ENSEMBL
366 ENSEMBL_MART_ENSEMBL
367 ENSEMBL_MART_ENSEMBL
417 ENSEMBL_MART_ENSEMBL
588 ENSEMBL_MART_ENSEMBL
600 ENSEMBL_MART_ENSEMBL
680 ENSEMBL_MART_ENSEMBL
681 ENSEMBL_MART_ENSEMBL
731 ENSEMBL_MART_ENSEMBL
902 ENSEMBL_MART_ENSEMBL
914 ENSEMBL_MART_ENSEMBL</code></pre>
<p>Now users can retrieve the corresponding GO attribute of <em>D.
rerio</em> with <code>organismAttributes</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># find go attribute term for D. rerio</span></span>
<span><span class="fu"><a href="../reference/organismAttributes.html">organismAttributes</a></span><span class="op">(</span><span class="st">"Danio rerio"</span>, topic <span class="op">=</span> <span class="st">"go"</span><span class="op">)</span></span></code></pre></div>
<pre><code>                                              name                             description
33                                           go_id                       GO Term Accession
36                                 go_linkage_type                   GO Term Evidence Code
38                            goslim_goa_accession                 GOSlim GOA Accession(s)
39                          goslim_goa_description                  GOSlim GOA Description
516                  ggorilla_homolog_ensembl_gene                 Gorilla Ensembl Gene ID
517  ggorilla_homolog_canomical_transcript_protein      Canonical Protein or Transcript ID
518               ggorilla_homolog_ensembl_peptide              Gorilla Ensembl Protein ID
519                    ggorilla_homolog_chromosome                 Gorilla Chromosome Name
520                   ggorilla_homolog_chrom_start           Gorilla Chromosome Start (bp)
521                     ggorilla_homolog_chrom_end             Gorilla Chromosome End (bp)
522                ggorilla_homolog_orthology_type                           Homology Type
523                       ggorilla_homolog_subtype                                Ancestor
524          ggorilla_homolog_orthology_confidence    Orthology confidence [0 low, 1 high]
525                       ggorilla_homolog_perc_id   % Identity with respect to query gene
526                    ggorilla_homolog_perc_id_r1 % Identity with respect to Gorilla gene
527                            ggorilla_homolog_dn                                      dN
528                            ggorilla_homolog_ds                                      dS
1240                                         go_id                                   GO ID
1241                                      quick_go                             Quick GO ID
1370                                         go_id                       GO Term Accession
1373                               go_linkage_type                   GO Term Evidence Code
1375                          goslim_goa_accession                 GOSlim GOA Accession(s)
1376                        goslim_goa_description                  GOSlim GOA Description
1853                 ggorilla_homolog_ensembl_gene                 Gorilla Ensembl Gene ID
1854 ggorilla_homolog_canomical_transcript_protein      Canonical Protein or Transcript ID
1855              ggorilla_homolog_ensembl_peptide              Gorilla Ensembl Protein ID
1856                   ggorilla_homolog_chromosome                 Gorilla Chromosome Name
1857                  ggorilla_homolog_chrom_start           Gorilla Chromosome Start (bp)
1858                    ggorilla_homolog_chrom_end             Gorilla Chromosome End (bp)
1859               ggorilla_homolog_orthology_type                           Homology Type
1860                      ggorilla_homolog_subtype                                Ancestor
1861         ggorilla_homolog_orthology_confidence    Orthology confidence [0 low, 1 high]
1862                      ggorilla_homolog_perc_id   % Identity with respect to query gene
1863                   ggorilla_homolog_perc_id_r1 % Identity with respect to Gorilla gene
1864                           ggorilla_homolog_dn                                      dN
1865                           ggorilla_homolog_ds                                      dS
2577                                         go_id                                   GO ID
2578                                      quick_go                             Quick GO ID
2707                                         go_id                       GO Term Accession
2710                               go_linkage_type                   GO Term Evidence Code
2712                          goslim_goa_accession                 GOSlim GOA Accession(s)
2713                        goslim_goa_description                  GOSlim GOA Description
3190                 ggorilla_homolog_ensembl_gene                 Gorilla Ensembl Gene ID
3191 ggorilla_homolog_canomical_transcript_protein      Canonical Protein or Transcript ID
3192              ggorilla_homolog_ensembl_peptide              Gorilla Ensembl Protein ID
3193                   ggorilla_homolog_chromosome                 Gorilla Chromosome Name
3194                  ggorilla_homolog_chrom_start           Gorilla Chromosome Start (bp)
3195                    ggorilla_homolog_chrom_end             Gorilla Chromosome End (bp)
3196               ggorilla_homolog_orthology_type                           Homology Type
3197                      ggorilla_homolog_subtype                                Ancestor
3198         ggorilla_homolog_orthology_confidence    Orthology confidence [0 low, 1 high]
3199                      ggorilla_homolog_perc_id   % Identity with respect to query gene
3200                   ggorilla_homolog_perc_id_r1 % Identity with respect to Gorilla gene
3201                           ggorilla_homolog_dn                                      dN
3202                           ggorilla_homolog_ds                                      dS
3914                                         go_id                                   GO ID
3915                                      quick_go                             Quick GO ID
                 dataset                 mart
33   drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
36   drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
38   drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
39   drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
516  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
517  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
518  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
519  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
520  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
521  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
522  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
523  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
524  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
525  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
526  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
527  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
528  drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1240    drerio_gene_vega ENSEMBL_MART_ENSEMBL
1241    drerio_gene_vega ENSEMBL_MART_ENSEMBL
1370 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1373 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1375 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1376 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1853 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1854 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1855 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1856 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1857 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1858 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1859 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1860 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1861 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1862 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1863 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1864 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
1865 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
2577    drerio_gene_vega ENSEMBL_MART_ENSEMBL
2578    drerio_gene_vega ENSEMBL_MART_ENSEMBL
2707 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
2710 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
2712 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
2713 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3190 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3191 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3192 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3193 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3194 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3195 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3196 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3197 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3198 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3199 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3200 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3201 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3202 drerio_gene_ensembl ENSEMBL_MART_ENSEMBL
3914    drerio_gene_vega ENSEMBL_MART_ENSEMBL
3915    drerio_gene_vega ENSEMBL_MART_ENSEMBL</code></pre>
<p>Now users can specify the filter <code>zfin_id</code> and attribute
<code>go_id</code> to retrieve the GO terms of corresponding gene ids
(<strong>Please note that this will take some time</strong>).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># retrieve GO terms of D. rerio brain genes originating in PS5</span></span>
<span><span class="va">GO_tbl.BrainGenes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/biomart.html">biomart</a></span><span class="op">(</span>genes      <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">BrainGenes.PS5</span><span class="op">[</span> , <span class="st">"ZFIN_ID"</span><span class="op">]</span><span class="op">)</span>,</span>
<span>                             mart       <span class="op">=</span> <span class="st">"ENSEMBL_MART_ENSEMBL"</span>,</span>
<span>                             dataset    <span class="op">=</span> <span class="st">"drerio_gene_ensembl"</span>,</span>
<span>                             attributes <span class="op">=</span> <span class="st">"go_id"</span>,</span>
<span>                             filters    <span class="op">=</span> <span class="st">"zfin_id"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO_tbl.BrainGenes</span><span class="op">)</span></span></code></pre></div>
<pre><code>            zfin_id      go_id
1 ZDB-GENE-000210-6 GO:0060037
2 ZDB-GENE-000210-6 GO:0046983
3 ZDB-GENE-000210-7 GO:0046983
4 ZDB-GENE-000328-4 GO:0007275
5 ZDB-GENE-000328-4 GO:0007166
6 ZDB-GENE-000328-4 GO:0035567</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
