<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>NCBI Database Retrieval • biomartr</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><meta property="og:title" content="NCBI Database Retrieval">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.8.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/biomartr">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>NCBI Database Retrieval</h1>
            
            <h4 class="date">2018-04-18</h4>
      
      <small>Source: <a href="https://github.com/ropensci/biomartr/blob/master/vignettes/Database_Retrieval.Rmd"><code>vignettes/Database_Retrieval.Rmd</code></a></small>
      <div class="hidden name"><code>Database_Retrieval.Rmd</code></div>

    </div>

    
    
<div id="retrieve-sequence-databases-from-ncbi" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieve-sequence-databases-from-ncbi" class="anchor"></a>Retrieve Sequence Databases from NCBI</h1>
<p>NCBI stores a variety of specialized database such as <a href="https://www.ncbi.nlm.nih.gov/guide/data-software/">Genbank, RefSeq, Taxonomy, SNP, etc.</a> on their servers. The <code><a href="../reference/download.database.html">download.database()</a></code> and <code><a href="../reference/download.database.all.html">download.database.all()</a></code> functions implemented in <code>biomartr</code> allows users to download these databases from NCBI.</p>
<div id="search-for-available-databases" class="section level2">
<h2 class="hasAnchor">
<a href="#search-for-available-databases" class="anchor"></a>Search for available databases</h2>
<p>Before downloading specific databases from NCBI users might want to list available databases. Using the <code><a href="../reference/listDatabases.html">listNCBIDatabases()</a></code> function users can retrieve a list of available databases stored on NCBI.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># retrieve a list of available sequence databases at NCBI</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"all"</span>)</code></pre></div>
<pre><code>[1] "16SMicrobial"            "cdd_delta"               "cloud"                  
 [4] "env_nr"                  "env_nt"                  "est"                    
 [7] "est_human"               "est_mouse"               "est_others"             
[10] "FASTA"                   "gene_info"               "gss"                    
[13] "gss_annot"               "htgs"                    "human_genomic"          
[16] "landmark"                "nr"                      "nt"                     
[19] "other_genomic"           "pataa"                   "patnt"                  
[22] "pdbaa"                   "pdbnt"                   "ref_prok_rep_genomes"   
[25] "ref_viroids_rep_genomes" "ref_viruses_rep_genomes" "refseq_genomic"         
[28] "refseq_protein"          "refseq_rna"              "refseqgene"             
[31] "sts"                     "swissprot"               "taxdb"                  
[34] "tsa_nr"                  "tsa_nt"                  "vector"</code></pre>
<p>However, in case users already know which database they would like to retrieve they can filter for the exact files by specifying the NCBI database name. In the following example all sequence files that are part of the <code>NCBI nr</code> database shall be retrieved.</p>
<p>First, the <code><a href="../reference/listDatabases.html">listNCBIDatabases(db = "nr")</a></code> allows to list all files corresponding to the <code>nr</code> database.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI nr files</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"nr"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "nr.00.tar.gz" "nr.83.tar.gz" "nr.75.tar.gz" "nr.01.tar.gz" "nr.02.tar.gz" "nr.03.tar.gz" "nr.04.tar.gz" "nr.05.tar.gz" "nr.76.tar.gz" "nr.16.tar.gz" "nr.54.tar.gz" "nr.55.tar.gz" "nr.56.tar.gz" "nr.06.tar.gz" "nr.15.tar.gz" "nr.30.tar.gz" "nr.57.tar.gz" "nr.58.tar.gz" "nr.59.tar.gz" "nr.60.tar.gz" "nr.61.tar.gz" "nr.62.tar.gz" "nr.63.tar.gz" "nr.64.tar.gz" "nr.65.tar.gz" "nr.66.tar.gz" "nr.67.tar.gz" "nr.68.tar.gz" "nr.69.tar.gz" "nr.70.tar.gz" "nr.71.tar.gz" "nr.72.tar.gz" "nr.73.tar.gz" "nr.74.tar.gz" "nr.07.tar.gz" "nr.77.tar.gz" "nr.78.tar.gz" "nr.08.tar.gz" "nr.09.tar.gz" "nr.10.tar.gz" "nr.79.tar.gz" "nr.80.tar.gz" "nr.81.tar.gz" "nr.82.tar.gz" "nr.84.tar.gz" "nr.11.tar.gz" "nr.12.tar.gz" "nr.13.tar.gz" "nr.14.tar.gz"
#&gt; [50] "nr.28.tar.gz" "nr.29.tar.gz" "nr.31.tar.gz" "nr.17.tar.gz" "nr.18.tar.gz" "nr.19.tar.gz" "nr.20.tar.gz" "nr.21.tar.gz" "nr.22.tar.gz" "nr.23.tar.gz" "nr.32.tar.gz" "nr.24.tar.gz" "nr.25.tar.gz" "nr.26.tar.gz" "nr.27.tar.gz" "nr.33.tar.gz" "nr.34.tar.gz" "nr.35.tar.gz" "nr.36.tar.gz" "nr.37.tar.gz" "nr.38.tar.gz" "nr.39.tar.gz" "nr.40.tar.gz" "nr.41.tar.gz" "nr.42.tar.gz" "nr.43.tar.gz" "nr.44.tar.gz" "nr.45.tar.gz" "nr.46.tar.gz" "nr.47.tar.gz" "nr.48.tar.gz" "nr.49.tar.gz" "nr.53.tar.gz" "nr.50.tar.gz" "nr.51.tar.gz" "nr.52.tar.gz"</code></pre>
<p>The output illustrates that the <code>NCBI nr</code> database has been separated into 41 files.</p>
<p>Further examples are:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI nt files</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"nt"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "nt.00.tar.gz" "nt.01.tar.gz" "nt.02.tar.gz" "nt.03.tar.gz" "nt.04.tar.gz" "nt.05.tar.gz" "nt.06.tar.gz" "nt.07.tar.gz" "nt.08.tar.gz" "nt.09.tar.gz" "nt.10.tar.gz" "nt.40.tar.gz" "nt.11.tar.gz" "nt.41.tar.gz" "nt.42.tar.gz" "nt.43.tar.gz" "nt.44.tar.gz" "nt.45.tar.gz" "nt.46.tar.gz" "nt.47.tar.gz" "nt.48.tar.gz" "nt.49.tar.gz" "nt.50.tar.gz" "nt.12.tar.gz" "nt.51.tar.gz" "nt.52.tar.gz" "nt.13.tar.gz" "nt.14.tar.gz" "nt.53.tar.gz" "nt.54.tar.gz" "nt.55.tar.gz" "nt.56.tar.gz" "nt.15.tar.gz" "nt.16.tar.gz" "nt.26.tar.gz" "nt.27.tar.gz" "nt.17.tar.gz" "nt.18.tar.gz" "nt.28.tar.gz" "nt.29.tar.gz" "nt.19.tar.gz" "nt.20.tar.gz" "nt.21.tar.gz" "nt.22.tar.gz" "nt.23.tar.gz" "nt.24.tar.gz" "nt.25.tar.gz" "nt.30.tar.gz" "nt.31.tar.gz"
#&gt; [50] "nt.32.tar.gz" "nt.33.tar.gz" "nt.34.tar.gz" "nt.35.tar.gz" "nt.36.tar.gz" "nt.37.tar.gz" "nt.39.tar.gz" "nt.38.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI ESTs others</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"est_others"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "est_others.00.tar.gz" "est_others.01.tar.gz" "est_others.02.tar.gz" "est_others.03.tar.gz" "est_others.04.tar.gz" "est_others.05.tar.gz" "est_others.06.tar.gz" "est_others.07.tar.gz" "est_others.08.tar.gz" "est_others.09.tar.gz" "est_others.10.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI RefSeq (only genomes)</span>
<span class="kw">head</span>(biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"refseq_genomic"</span>), <span class="dv">20</span>)</code></pre></div>
<pre><code>#&gt;  [1] "refseq_genomic.218.tar.gz" "refseq_genomic.219.tar.gz" "refseq_genomic.220.tar.gz" "refseq_genomic.00.tar.gz"  "refseq_genomic.01.tar.gz"  "refseq_genomic.02.tar.gz"  "refseq_genomic.03.tar.gz"  "refseq_genomic.04.tar.gz"  "refseq_genomic.05.tar.gz"  "refseq_genomic.06.tar.gz"  "refseq_genomic.07.tar.gz"  "refseq_genomic.08.tar.gz"  "refseq_genomic.09.tar.gz"  "refseq_genomic.10.tar.gz"  "refseq_genomic.11.tar.gz"  "refseq_genomic.12.tar.gz"  "refseq_genomic.13.tar.gz"  "refseq_genomic.14.tar.gz"  "refseq_genomic.15.tar.gz"  "refseq_genomic.16.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI RefSeq (only proteomes)</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"refseq_protein"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "refseq_protein.25.tar.gz" "refseq_protein.00.tar.gz" "refseq_protein.01.tar.gz" "refseq_protein.02.tar.gz" "refseq_protein.03.tar.gz" "refseq_protein.26.tar.gz" "refseq_protein.27.tar.gz" "refseq_protein.28.tar.gz" "refseq_protein.29.tar.gz" "refseq_protein.04.tar.gz" "refseq_protein.05.tar.gz" "refseq_protein.06.tar.gz" "refseq_protein.07.tar.gz" "refseq_protein.08.tar.gz" "refseq_protein.09.tar.gz" "refseq_protein.14.tar.gz" "refseq_protein.15.tar.gz" "refseq_protein.16.tar.gz" "refseq_protein.10.tar.gz" "refseq_protein.11.tar.gz" "refseq_protein.17.tar.gz" "refseq_protein.12.tar.gz" "refseq_protein.13.tar.gz" "refseq_protein.18.tar.gz" "refseq_protein.19.tar.gz" "refseq_protein.20.tar.gz" "refseq_protein.21.tar.gz"
#&gt; [28] "refseq_protein.22.tar.gz" "refseq_protein.23.tar.gz" "refseq_protein.24.tar.gz" "refseq_protein.30.tar.gz" "refseq_protein.31.tar.gz" "refseq_protein.32.tar.gz" "refseq_protein.33.tar.gz" "refseq_protein.34.tar.gz" "refseq_protein.35.tar.gz" "refseq_protein.36.tar.gz" "refseq_protein.37.tar.gz" "refseq_protein.38.tar.gz" "refseq_protein.39.tar.gz" "refseq_protein.40.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show all NCBI RefSeq (only RNA)</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"refseq_rna"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "refseq_rna.00.tar.gz" "refseq_rna.01.tar.gz" "refseq_rna.10.tar.gz" "refseq_rna.02.tar.gz" "refseq_rna.05.tar.gz" "refseq_rna.03.tar.gz" "refseq_rna.06.tar.gz" "refseq_rna.04.tar.gz" "refseq_rna.07.tar.gz" "refseq_rna.08.tar.gz" "refseq_rna.09.tar.gz" "refseq_rna.11.tar.gz" "refseq_rna.12.tar.gz" "refseq_rna.13.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show NCBI swissprot</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"swissprot"</span>)</code></pre></div>
<pre><code>#&gt; [1] "swissprot.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show NCBI PDB</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"pdb"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "pdbnt.00.tar.gz" "pdbnt.40.tar.gz" "pdbnt.41.tar.gz" "pdbnt.42.tar.gz" "pdbnt.43.tar.gz" "pdbnt.44.tar.gz" "pdbnt.45.tar.gz" "pdbnt.46.tar.gz" "pdbnt.47.tar.gz" "pdbnt.48.tar.gz" "pdbnt.49.tar.gz" "pdbnt.50.tar.gz" "pdbnt.51.tar.gz" "pdbnt.52.tar.gz" "pdbnt.53.tar.gz" "pdbnt.54.tar.gz" "pdbnt.55.tar.gz" "pdbnt.56.tar.gz" "pdbnt.26.tar.gz" "pdbnt.27.tar.gz" "pdbnt.01.tar.gz" "pdbnt.02.tar.gz" "pdbnt.03.tar.gz" "pdbnt.04.tar.gz" "pdbnt.05.tar.gz" "pdbnt.06.tar.gz" "pdbnt.07.tar.gz" "pdbnt.08.tar.gz" "pdbnt.09.tar.gz" "pdbnt.10.tar.gz" "pdbnt.11.tar.gz" "pdbnt.12.tar.gz" "pdbnt.13.tar.gz" "pdbnt.14.tar.gz" "pdbnt.15.tar.gz" "pdbnt.16.tar.gz" "pdbnt.17.tar.gz" "pdbnt.18.tar.gz" "pdbnt.28.tar.gz" "pdbnt.19.tar.gz" "pdbnt.20.tar.gz"
#&gt; [42] "pdbnt.21.tar.gz" "pdbnt.22.tar.gz" "pdbnt.23.tar.gz" "pdbnt.24.tar.gz" "pdbnt.29.tar.gz" "pdbnt.25.tar.gz" "pdbaa.tar.gz"    "pdbnt.30.tar.gz" "pdbnt.31.tar.gz" "pdbnt.32.tar.gz" "pdbnt.33.tar.gz" "pdbnt.34.tar.gz" "pdbnt.35.tar.gz" "pdbnt.36.tar.gz" "pdbnt.37.tar.gz" "pdbnt.39.tar.gz" "pdbnt.38.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show NCBI Human database</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"human"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "human_genomic.00.tar.gz" "human_genomic.01.tar.gz" "human_genomic.02.tar.gz" "human_genomic.03.tar.gz" "human_genomic.04.tar.gz" "human_genomic.05.tar.gz" "human_genomic.06.tar.gz" "human_genomic.07.tar.gz" "human_genomic.08.tar.gz" "human_genomic.10.tar.gz" "human_genomic.11.tar.gz" "human_genomic.12.tar.gz" "human_genomic.13.tar.gz" "human_genomic.14.tar.gz" "human_genomic.15.tar.gz" "human_genomic.16.tar.gz" "human_genomic.18.tar.gz" "human_genomic.19.tar.gz" "human_genomic.20.tar.gz" "human_genomic.21.tar.gz" "human_genomic.22.tar.gz"</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># show NCBI EST databases</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/listDatabases">listNCBIDatabases</a></span>(<span class="dt">db =</span> <span class="st">"est"</span>)</code></pre></div>
<pre><code>#&gt;  [1] "est.tar.gz"           "est_human.00.tar.gz"  "est_human.01.tar.gz"  "est_mouse.tar.gz"     "est_others.00.tar.gz" "est_others.01.tar.gz" "est_others.02.tar.gz" "est_others.03.tar.gz" "est_others.04.tar.gz" "est_others.05.tar.gz" "est_others.06.tar.gz" "est_others.07.tar.gz" "est_others.08.tar.gz" "est_others.09.tar.gz" "est_others.10.tar.gz"</code></pre>
<p><strong>Please not that all lookup and retrieval function will only work properly when a sufficient internet connection is provided.</strong></p>
<p>In a next step users can use the <code><a href="../reference/listDatabases.html">listNCBIDatabases()</a></code> and <code><a href="../reference/download.database.all.html">download.database.all()</a></code> functions to retrieve all files corresponding to a specific NCBI database.</p>
</div>
<div id="download-available-databases" class="section level2">
<h2 class="hasAnchor">
<a href="#download-available-databases" class="anchor"></a>Download available databases</h2>
<p>Using the same search strategy by specifying the database name as described above, users can now download these databases using the <code><a href="../reference/download.database.html">download.database()</a></code> function.</p>
<p>For downloading only single files users can type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># select NCBI nr version 27 =  "nr.27.tar.gz"</span>
<span class="co"># and download it into the folder</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/download.database">download.database</a></span>(<span class="dt">db =</span> <span class="st">"nr.27.tar.gz"</span>, <span class="dt">path =</span> <span class="st">"nr"</span>)</code></pre></div>
<p>Using this command first a folder named <code>nr</code> is created and the file <code>nr.27.tar.gz</code> is downloaded to this folder. This command will download the pre-formatted (by makeblastdb formatted) database version is retrieved.</p>
<p>Alternatively, users can retrieve all <code>nr</code> files with one command by typing:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download the entire NCBI nr database</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/download.database.all">download.database.all</a></span>(<span class="dt">db =</span> <span class="st">"nr"</span>, <span class="dt">path =</span> <span class="st">"nr"</span>)</code></pre></div>
<p>Using this command, all <code>NCBI nr</code> files are loaded into the <code>nr</code> folder (<code>path = "nr"</code>). For each data package, <code>biomartr</code> checks the <code>md5checksum</code> of the downloaded file and the file stored online to make sure that internet connection losses didn’t currupt the file. In case you see a warning message notifying you about not-matching <code>md5checksum</code> values, please re-download the corresponding data package. From my own experience this can happen when server connections or internet connections are not very stable during the download process of large data chunks.</p>
<p>The same approach can be applied to all other databases mentioned above, e.g.:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download the entire NCBI nt database</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/download.database.all">download.database.all</a></span>(<span class="dt">db =</span> <span class="st">"nt"</span>, <span class="dt">path =</span> <span class="st">"nt"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download the entire NCBI refseq (protein) database</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/download.database.all">download.database.all</a></span>(<span class="dt">db =</span> <span class="st">"refseq_protein"</span>, <span class="dt">path =</span> <span class="st">"refseq_protein"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># download the entire NCBI PDB database</span>
biomartr<span class="op">::</span><span class="kw"><a href="http://www.rdocumentation.org/packages/biomartr/topics/download.database.all">download.database.all</a></span>(<span class="dt">db =</span> <span class="st">"pdb"</span>, <span class="dt">path =</span> <span class="st">"pdb"</span>)</code></pre></div>
<p>For each data package, <code>biomartr</code> checks the <code>md5checksum</code> of the downloaded file and the file stored online to make sure that internet connection losses didn’t currupt the file. In case you see a warning message notifying you about not-matching <code>md5checksum</code> values, please re-download the corresponding data package. From my own experience this can happen when server connections or internet connections are not very stable during the download process of large data chunks.</p>
<p><strong>Please notice that most of these databases are very large, so users should take of of providing a stable internet connection throughout the download process.</strong></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#retrieve-sequence-databases-from-ncbi">Retrieve Sequence Databases from NCBI</a><ul class="nav nav-pills nav-stacked">
<li><a href="#search-for-available-databases">Search for available databases</a></li>
      <li><a href="#download-available-databases">Download available databases</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  </body>
</html>
