<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>NCBI Database Retrieval • biomartr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="NCBI Database Retrieval">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>NCBI Database Retrieval</h1>
            
            <h4 data-toc-skip class="date">2025-07-19</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/vignettes/Database_Retrieval.Rmd" class="external-link"><code>vignettes/Database_Retrieval.Rmd</code></a></small>
      <div class="hidden name"><code>Database_Retrieval.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="retrieve-sequence-databases-from-ncbi">Retrieve Sequence Databases from NCBI<a class="anchor" aria-label="anchor" href="#retrieve-sequence-databases-from-ncbi"></a>
</h2>
<blockquote>
<p><strong><em>NOTE:</em></strong> To make sure that you have a
sufficiently stable (internet) connection between R and the respective
databases, please set the default <code>timeout</code> setting
<strong>on your local machine</strong> from 60sec to at least 300000sec
before running any retrieval functions via:</p>
</blockquote>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>timeout <span class="op">=</span> <span class="fl">300000</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
<p>NCBI stores a variety of specialized database such as <a href="https://www.ncbi.nlm.nih.gov/guide/data-software/" class="external-link">Genbank,
RefSeq, Taxonomy, SNP, etc.</a> on their servers. The
<code><a href="../reference/download.database.html">download.database()</a></code> and
<code><a href="../reference/download.database.all.html">download.database.all()</a></code> functions implemented in
<code>biomartr</code> allows users to download these databases from
NCBI. This process might be very useful for downstream analyses such as
sequence searches with e.g. BLAST. For this purpose see the R package <a href="https://github.com/drostlab/metablastr" class="external-link">metablastr</a> which aims
to seamlessly integrate <code>biomartr</code> based genomic data
retrieval with downstream large-scale BLAST searches.</p>
<ul>
<li><a href="#ist-available-databases">1. List available NCBI databases
with <code>listNCBIDatabases()</code></a></li>
<li>
<a href="#download-ncbi-databases">2. Download NCBI databases with
<code>download.database.all()</code></a>
<ul>
<li><a href="#example-ncbi-nr">2.1 Example NCBI nr retrieval</a></li>
<li><a href="#example-ncbi-nt">2.2 Example NCBI nt retrieval</a></li>
<li><a href="#example-ncbi-refseq">2.3 Example NCBI RefSeq
retrieval</a></li>
<li><a href="#example-pdb">2.4 Example Protein Database (PDB)
retrieval</a></li>
<li><a href="#example-ncbi-taxonomy">2.5 Example NCBI Taxonomy
retrieval</a></li>
<li><a href="#example-ncbi-swissprot">2.6 Example NCBI Swissprot
retrieval</a></li>
<li><a href="#example-ncbi-cdd-delta">2.7 Example NCBI CDD Delta
retrieval</a></li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="list-available-databases">List available databases<a class="anchor" aria-label="anchor" href="#list-available-databases"></a>
</h3>
<p>Before downloading specific databases from NCBI users might want to
list available databases. Using the <code><a href="../reference/listDatabases.html">listNCBIDatabases()</a></code>
function users can retrieve a list of available databases stored on
NCBI.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># retrieve a list of available sequence databases at NCBI</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "16S_ribosomal_RNA"                     "16S_ribosomal_RNA-nucl-metadata"      
 [3] "18S_fungal_sequences"                  "18S_fungal_sequences-nucl-metadata"   
 [5] "28S_fungal_sequences"                  "28S_fungal_sequences-nucl-metadata"   
 [7] "Betacoronavirus"                       "Betacoronavirus-nucl-metadata"        
 [9] "blastdb-manifest"                      "blastdb-metadata-1-1"                 
[11] "blastdbv5"                             "cdd_delta"                            
[13] "cloud"                                 "env_nr"                               
[15] "env_nr-prot-metadata"                  "env_nt"                               
[17] "env_nt-nucl-metadata"                  "FASTA"                                
[19] "human_genome"                          "human_genome-nucl-metadata"           
[21] "ITS_eukaryote_sequences"               "ITS_eukaryote_sequences-nucl-metadata"
[23] "ITS_RefSeq_Fungi"                      "ITS_RefSeq_Fungi-nucl-metadata"       
[25] "landmark"                              "landmark-prot-metadata"               
[27] "LSU_eukaryote_rRNA"                    "LSU_eukaryote_rRNA-nucl-metadata"     
[29] "LSU_prokaryote_rRNA"                   "LSU_prokaryote_rRNA-nucl-metadata"    
[31] "mito"                                  "mito-nucl-metadata"                   
[33] "mouse_genome"                          "mouse_genome-nucl-metadata"           
[35] "nr"                                    "nr-prot-metadata"                     
[37] "nt"                                    "nt-nucl-metadata"                     
[39] "pataa"                                 "pataa-prot-metadata"                  
[41] "patnt"                                 "patnt-nucl-metadata"                  
[43] "pdbaa"                                 "pdbaa-prot-metadata"                  
[45] "pdbnt"                                 "pdbnt-nucl-metadata"                  
[47] "ref_euk_rep_genomes"                   "ref_euk_rep_genomes-nucl-metadata"    
[49] "ref_prok_rep_genomes"                  "ref_prok_rep_genomes-nucl-metadata"   
[51] "ref_viroids_rep_genomes"               "ref_viroids_rep_genomes-nucl-metadata"
[53] "ref_viruses_rep_genomes"               "ref_viruses_rep_genomes-nucl-metadata"
[55] "refseq_protein"                        "refseq_protein-prot-metadata"         
[57] "refseq_rna"                            "refseq_rna-nucl-metadata"             
[59] "refseq_select_prot"                    "refseq_select_prot-prot-metadata"     
[61] "refseq_select_rna"                     "refseq_select_rna-nucl-metadata"      
[63] "SSU_eukaryote_rRNA"                    "SSU_eukaryote_rRNA-nucl-metadata"     
[65] "swissprot"                             "swissprot-prot-metadata"              
[67] "taxdb"                                 "taxdb-metadata"                       
[69] "tsa_nr"                                "tsa_nr-prot-metadata"                 
[71] "tsa_nt"                                "tsa_nt-nucl-metadata"                 
[73] "v4"                                    "v5" </code></pre>
<p>However, in case users already know which database they would like to
retrieve they can filter for the exact files by specifying the NCBI
database name. In the following example all sequence files that are part
of the <code>NCBI nr</code> database shall be retrieved.</p>
<p>First, the <code>listNCBIDatabases(db = "nr")</code> allows to list
all files corresponding to the <code>nr</code> database.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show all NCBI nr files</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"nr"</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "nr.00.tar.gz"          "nr.01.tar.gz"         
 [3] "nr.02.tar.gz"          "nr.03.tar.gz"         
 [5] "nr.04.tar.gz"          "nr.05.tar.gz"         
 [7] "nr.06.tar.gz"          "nr.07.tar.gz"         
 [9] "nr.08.tar.gz"          "nr.09.tar.gz"         
[11] "nr.10.tar.gz"          "nr.11.tar.gz"         
[13] "nr.12.tar.gz"          "nr.13.tar.gz"         
[15] "nr.14.tar.gz"          "nr.15.tar.gz"         
[17] "nr.16.tar.gz"          "nr.17.tar.gz"         
[19] "nr.18.tar.gz"          "nr.19.tar.gz"         
[21] "nr.20.tar.gz"          "nr.21.tar.gz"         
[23] "nr.22.tar.gz"          "nr.23.tar.gz"         
[25] "nr.24.tar.gz"          "nr.25.tar.gz"         
[27] "nr.26.tar.gz"          "nr.27.tar.gz"         
[29] "nr.28.tar.gz"          "nr.29.tar.gz"         
[31] "nr.30.tar.gz"          "nr.31.tar.gz"         
[33] "nr.32.tar.gz"          "nr.33.tar.gz"         
[35] "nr.34.tar.gz"          "nr.35.tar.gz"         
[37] "nr.36.tar.gz"          "nr.37.tar.gz"         
[39] "nr.38.tar.gz"          "nr.39.tar.gz"         
[41] "nr.40.tar.gz"          "nr.41.tar.gz"         
[43] "nr.42.tar.gz"          "nr.43.tar.gz"         
[45] "nr.44.tar.gz"          "nr.45.tar.gz"         
[47] "nr.46.tar.gz"          "nr-prot-metadata.json"
[49] "nr.47.tar.gz"          "nr.48.tar.gz"         
[51] "nr.49.tar.gz"          "nr.50.tar.gz"         
[53] "nr.51.tar.gz"          "nr.52.tar.gz"         
[55] "nr.53.tar.gz"          "nr.54.tar.gz"         
[57] "nr.55.tar.gz"</code></pre>
<p>The output illustrates that the <code>NCBI nr</code> database has
been separated into several sub-data-packages.</p>
<p>Further examples are:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show all NCBI nt files</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"nt"</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "nt.00.tar.gz"          "nt.01.tar.gz"         
 [3] "nt.02.tar.gz"          "nt.03.tar.gz"         
 [5] "nt.04.tar.gz"          "nt.05.tar.gz"         
 [7] "nt.06.tar.gz"          "nt.07.tar.gz"         
 [9] "nt.08.tar.gz"          "nt.09.tar.gz"         
[11] "nt.10.tar.gz"          "nt.11.tar.gz"         
[13] "nt.12.tar.gz"          "nt.13.tar.gz"         
[15] "nt.14.tar.gz"          "nt.15.tar.gz"         
[17] "nt.16.tar.gz"          "nt.17.tar.gz"         
[19] "nt.18.tar.gz"          "nt.19.tar.gz"         
[21] "nt.20.tar.gz"          "nt.21.tar.gz"         
[23] "nt.22.tar.gz"          "nt.23.tar.gz"         
[25] "nt.24.tar.gz"          "nt.25.tar.gz"         
[27] "nt.26.tar.gz"          "nt.27.tar.gz"         
[29] "nt.28.tar.gz"          "nt.29.tar.gz"         
[31] "nt.30.tar.gz"          "nt.31.tar.gz"         
[33] "nt.32.tar.gz"          "nt.33.tar.gz"         
[35] "nt.34.tar.gz"          "nt.35.tar.gz"         
[37] "nt.36.tar.gz"          "nt.37.tar.gz"         
[39] "nt-nucl-metadata.json" "nt.38.tar.gz"         
[41] "nt.39.tar.gz"          "nt.40.tar.gz"         
[43] "nt.41.tar.gz"          "nt.42.tar.gz"         
[45] "nt.43.tar.gz"          "nt.44.tar.gz"         
[47] "nt.45.tar.gz"          "nt.46.tar.gz"         
[49] "nt.47.tar.gz"          "nt.48.tar.gz"</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show all NCBI RefSeq (only proteomes)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq_protein"</span><span class="op">)</span>, <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "refseq_protein.00.tar.gz" "refseq_protein.01.tar.gz"
 [3] "refseq_protein.02.tar.gz" "refseq_protein.03.tar.gz"
 [5] "refseq_protein.04.tar.gz" "refseq_protein.05.tar.gz"
 [7] "refseq_protein.06.tar.gz" "refseq_protein.07.tar.gz"
 [9] "refseq_protein.08.tar.gz" "refseq_protein.09.tar.gz"
[11] "refseq_protein.10.tar.gz" "refseq_protein.11.tar.gz"
[13] "refseq_protein.12.tar.gz" "refseq_protein.13.tar.gz"
[15] "refseq_protein.14.tar.gz" "refseq_protein.15.tar.gz"
[17] "refseq_protein.16.tar.gz" "refseq_protein.17.tar.gz"
[19] "refseq_protein.18.tar.gz" "refseq_protein.19.tar.gz"</code></pre>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show all NCBI RefSeq (only RNA)</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq_rna"</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "refseq_rna.00.tar.gz"          "refseq_rna.01.tar.gz"         
 [3] "refseq_rna.02.tar.gz"          "refseq_rna.03.tar.gz"         
 [5] "refseq_rna.04.tar.gz"          "refseq_rna.05.tar.gz"         
 [7] "refseq_rna.06.tar.gz"          "refseq_rna.07.tar.gz"         
 [9] "refseq_rna.08.tar.gz"          "refseq_rna-nucl-metadata.json"
[11] "refseq_rna.09.tar.gz" </code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show NCBI swissprot</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"swissprot"</span><span class="op">)</span></span></code></pre></div>
<pre><code>[1] "swissprot.tar.gz"             "swissprot-prot-metadata.json"</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show NCBI PDB</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"pdb"</span><span class="op">)</span></span></code></pre></div>
<pre><code> [1] "pdbaa.tar.gz"             "pdbnt.tar.gz"            
[3] "pdbaa-prot-metadata.json" "pdbnt-nucl-metadata.json"</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># show NCBI Human database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/listDatabases.html">listNCBIDatabases</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"human"</span><span class="op">)</span></span></code></pre></div>
<pre><code>1] "human_genome.00.tar.gz"          "human_genome.01.tar.gz"         
[3] "human_genome-nucl-metadata.json"</code></pre>
<p><strong>Please not that all lookup and retrieval function will only
work properly when a sufficient internet connection is
provided.</strong></p>
<p>In a next step users can use the <code><a href="../reference/listDatabases.html">listNCBIDatabases()</a></code> and
<code><a href="../reference/download.database.all.html">download.database.all()</a></code> functions to retrieve all files
corresponding to a specific NCBI database.</p>
</div>
<div class="section level3">
<h3 id="download-ncbi-databases">Download NCBI databases<a class="anchor" aria-label="anchor" href="#download-ncbi-databases"></a>
</h3>
<p>Using the same search strategy by specifying the database name as
described above, users can now download these databases using the
<code><a href="../reference/download.database.all.html">download.database.all()</a></code> function.</p>
<p>For downloading only single files users can type:</p>
<div class="section level4">
<h4 id="example-ncbi-nr">Example NCBI nr<a class="anchor" aria-label="anchor" href="#example-ncbi-nr"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI nr database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"nr"</span>, path <span class="op">=</span> <span class="st">"nr"</span><span class="op">)</span></span></code></pre></div>
<p>This command will download the pre-formatted (by makeblastdb
formatted) database version is retrieved.</p>
<p>Using this command, all <code>NCBI nr</code> files are loaded into
the <code>nr</code> folder (<code>path = "nr"</code>). For each data
package, <code>biomartr</code> checks the <code>md5checksum</code> of
the downloaded file and the file stored online to make sure that
internet connection losses didn’t currupt the file. In case you see a
warning message notifying you about not-matching
<code>md5checksum</code> values, please re-download the corresponding
data package by re-running the <code><a href="../reference/download.database.all.html">download.database.all()</a></code>
command. From my own experience this can happen when server connections
or internet connections are not very stable during the download process
of large data chunks.</p>
</div>
<div class="section level4">
<h4 id="example-ncbi-nt">Example NCBI nt<a class="anchor" aria-label="anchor" href="#example-ncbi-nt"></a>
</h4>
<p>The same approach can be applied to all other databases mentioned
above, e.g.:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI nt database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"nt"</span>, path <span class="op">=</span> <span class="st">"nt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-ncbi-refseq">Example NCBI RefSeq<a class="anchor" aria-label="anchor" href="#example-ncbi-refseq"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI refseq (protein) database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq_protein"</span>, path <span class="op">=</span> <span class="st">"refseq_protein"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-pdb">Example PDB<a class="anchor" aria-label="anchor" href="#example-pdb"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI PDB database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"pdb"</span>, path <span class="op">=</span> <span class="st">"pdb"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-ncbi-taxonomy">Example NCBI Taxonomy<a class="anchor" aria-label="anchor" href="#example-ncbi-taxonomy"></a>
</h4>
<p>Download NCBI Taxonomy via:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI taxonomy database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"taxdb"</span>, path <span class="op">=</span> <span class="st">"taxdb"</span><span class="op">)</span></span></code></pre></div>
<pre><code>Starting download of the files: taxdb.tar.gz, taxdb.btd, taxdb.bti ...
This download process may take a while due to the large size of the individual data chunks ...
Starting download process of file: taxdb.tar.gz ...
Checking md5 hash of file: taxdb.tar.gz ...
The md5 hash of file 'taxdb.tar.gz' matches!
File 'taxdb/taxdb.tar.gz has successfully been retrieved.
Starting download process of file: taxdb.btd ...
Checking md5 hash of file: taxdb.btd ...
The md5 hash of file 'taxdb.btd' matches!
File 'taxdb/taxdb.btd has successfully been retrieved.
Starting download process of file: taxdb.bti ...
Checking md5 hash of file: taxdb.bti ...
The md5 hash of file 'taxdb.bti' matches!
File 'taxdb/taxdb.bti has successfully been retrieved.
Download process is finished and files are stored in 'taxdb'.</code></pre>
</div>
<div class="section level4">
<h4 id="example-ncbi-swissprot">Example NCBI Swissprot<a class="anchor" aria-label="anchor" href="#example-ncbi-swissprot"></a>
</h4>
<p>Download NCBI Swissprot via:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI swissprot database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"swissprot"</span>, path <span class="op">=</span> <span class="st">"swissprot"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="example-ncbi-cdd-delta">Example NCBI CDD Delta<a class="anchor" aria-label="anchor" href="#example-ncbi-cdd-delta"></a>
</h4>
<p>Download NCBI CDD Delta via:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the entire NCBI CDD Delta database</span></span>
<span><span class="fu">biomartr</span><span class="fu">::</span><span class="fu"><a href="../reference/download.database.all.html">download.database.all</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"cdd_delta"</span>, path <span class="op">=</span> <span class="st">"cdd_delta"</span><span class="op">)</span></span></code></pre></div>
<p>For each data package, <code>biomartr</code> checks the
<code>md5checksum</code> of the downloaded file and the file stored
online to make sure that internet connection losses didn’t currupt the
file. In case you see a warning message notifying you about not-matching
<code>md5checksum</code> values, please re-download the corresponding
data package. From my own experience this can happen when server
connections or internet connections are not very stable during the
download process of large data chunks.</p>
<p><strong>Please notice that most of these databases are very large, so
users should take of of providing a stable internet connection
throughout the download process.</strong></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
