<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>GFF retrieval of multiple species — getGFFSet • biomartr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="GFF retrieval of multiple species — getGFFSet"><meta property="og:description" content="Main GFF retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the GFF of the organisms of interest
will be downloaded and stored locally. GFF files can be retrieved from several databases."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>GFF retrieval of multiple species</h1>
    <small class="dont-index">Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/R/getSet.R" class="external-link"><code>R/getSet.R</code></a></small>
    <div class="hidden name"><code>getGFFSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Main GFF retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the GFF of the organisms of interest
will be downloaded and stored locally. GFF files can be retrieved from several databases.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getGFFSet</span><span class="op">(</span></span>
<span>  db <span class="op">=</span> <span class="st">"refseq"</span>,</span>
<span>  <span class="va">organisms</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  release <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  skip_bacteria <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  gunzip <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  remove_annotation_outliers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  update <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  format <span class="op">=</span> <span class="st">"gff3"</span>,</span>
<span>  path <span class="op">=</span> <span class="st">"set_GFF"</span>,</span>
<span>  mute_citation <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-db">db<a class="anchor" aria-label="anchor" href="#arg-db"></a></dt>
<dd><p>a character string specifying the database from which the genome
shall be retrieved:</p><ul><li><p><code>db = "refseq"</code></p></li>
<li><p><code>db = "genbank"</code></p></li>
<li><p><code>db = "ensembl"</code></p></li>
</ul></dd>


<dt id="arg-organisms">organisms<a class="anchor" aria-label="anchor" href="#arg-organisms"></a></dt>
<dd><p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p></dd>


<dt id="arg-reference">reference<a class="anchor" aria-label="anchor" href="#arg-reference"></a></dt>
<dd><p>a logical value indicating whether or not a genome shall be a candidate for downloaded
if it isn't marked in the database as either a reference genome or a representative genome.
This is helpful if you don't want to allow "partial genomes" etc.</p></dd>


<dt id="arg-release">release<a class="anchor" aria-label="anchor" href="#arg-release"></a></dt>
<dd><p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p></dd>


<dt id="arg-skip-bacteria">skip_bacteria<a class="anchor" aria-label="anchor" href="#arg-skip-bacteria"></a></dt>
<dd><p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p></dd>


<dt id="arg-gunzip">gunzip<a class="anchor" aria-label="anchor" href="#arg-gunzip"></a></dt>
<dd><p>a logical, indicating whether or not files should be unzipped.</p></dd>


<dt id="arg-remove-annotation-outliers">remove_annotation_outliers<a class="anchor" aria-label="anchor" href="#arg-remove-annotation-outliers"></a></dt>
<dd><p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></code>
for further exploration.</p></dd>


<dt id="arg-update">update<a class="anchor" aria-label="anchor" href="#arg-update"></a></dt>
<dd><p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p></dd>


<dt id="arg-format">format<a class="anchor" aria-label="anchor" href="#arg-format"></a></dt>
<dd><p>"gff3", alternative "gtf" for ensembl.</p></dd>


<dt id="arg-path">path<a class="anchor" aria-label="anchor" href="#arg-path"></a></dt>
<dd><p>character, default location is paste0("set_", toupper(set_type))</p></dd>


<dt id="arg-mute-citation">mute_citation<a class="anchor" aria-label="anchor" href="#arg-mute-citation"></a></dt>
<dd><p>logical, default FALSE, indicating whether citation message should be muted.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>character vector, the file path to the downloaded genomes,<br>
The returned character vector has names as either:<br>
- 'new' (file was downloaded now)<br>
- 'old' files did already exist)</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Fetching of assembly / sequence data is done by fetching an overview file
from metadata of given database:<br>
For NCBI (refseq/genbank): <br>
Internally this function loads the the overview.txt file from NCBI:</p>
<p>refseq: ftp.ncbi.nlm.nih.gov/genomes/refseq/</p>
<p>genbank: ftp.ncbi.nlm.nih.gov/genomes/genbank/</p>
<p>It will then create a directory relative to file type wanted,
if you get fasta genomes it will be
_ncbi_downloads/genomes' etc.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.<br></p>
<p>For ensembl it fetches overview per type from the rest API:</p>
<p>ensembl: https://rest.ensembl.org</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other getBioSet:
<code><a href="getBioSet.html">getBioSet</a>()</code>,
<code><a href="getCDSSet.html">getCDSSet</a>()</code>,
<code><a href="getCollectionSet.html">getCollectionSet</a>()</code>,
<code><a href="getGenomeSet.html">getGenomeSet</a>()</code>,
<code><a href="getProteomeSet.html">getProteomeSet</a>()</code>,
<code><a href="getRNASet.html">getRNASet</a>()</code></p>
<p>Other gff:
<code><a href="getGFF.html">getGFF</a>()</code>,
<code><a href="read_gff.html">read_gff</a>()</code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="getBioSet.html">getBioSet</a></span><span class="op">(</span><span class="st">"refseq"</span>, organisms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Arabidopsis thaliana"</span>,</span></span>
<span class="r-in"><span>                                  <span class="st">"Arabidopsis lyrata"</span>,</span></span>
<span class="r-in"><span>                                  <span class="st">"Capsella rubella"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                  set_type <span class="op">=</span> <span class="st">"cds"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

