<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Proteome Retrieval — getProteome • biomartr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Proteome Retrieval — getProteome"><meta property="og:description" content="Main proteome retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the proteome of the organism of interest
can be downloaded and stored locally. Proteome files can be retrieved from
several databases."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.7.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Proteome Retrieval</h1>
    <small class="dont-index">Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/R/getProteome.R" class="external-link"><code>R/getProteome.R</code></a></small>
    <div class="hidden name"><code>getProteome.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Main proteome retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the proteome of the organism of interest
can be downloaded and stored locally. Proteome files can be retrieved from
several databases.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getProteome</span><span class="op">(</span></span>
<span>  db <span class="op">=</span> <span class="st">"refseq"</span>,</span>
<span>  <span class="va">organism</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  skip_bacteria <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  release <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gunzip <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  update <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"_ncbi_downloads"</span>, <span class="st">"proteomes"</span><span class="op">)</span>,</span>
<span>  mute_citation <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>db</dt>
<dd><p>a character string specifying the database from which the genome
shall be retrieved:</p><ul><li><p><code>db = "refseq"</code></p></li>
<li><p><code>db = "genbank"</code></p></li>
<li><p><code>db = "ensembl"</code></p></li>
</ul></dd>


<dt>organism</dt>
<dd><p>Organism selector id,
there are three options to characterize an organism:</p><ul><li><p>by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code></p></li>
<li><p>by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)</p></li>
<li><p>by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)</p></li>
</ul></dd>


<dt>reference</dt>
<dd><p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p></dd>


<dt>skip_bacteria</dt>
<dd><p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p></dd>


<dt>release</dt>
<dd><p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p></dd>


<dt>gunzip</dt>
<dd><p>a logical, indicating whether or not files should be unzipped.</p></dd>


<dt>update</dt>
<dd><p>logical, default TRUE. (Uniprot only for now!)
If species info file exists already,
do not re download, makes it faster but the file can be old, i.e. no longer
as complete as it could be.</p></dd>


<dt>path</dt>
<dd><p>a character string specifying the location (a folder) in which
the corresponding proteome shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","proteomes")</code>.</p></dd>


<dt>mute_citation</dt>
<dd><p>logical, default FALSE, indicating whether citation message should be muted.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>File path to downloaded proteome.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Internally this function loads the the overview.txt file from NCBI:</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/</p>
<p>and creates a directory '_ncbi_downloads/proteomes' to store
the proteome of interest as fasta file for future processing.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other getBio: 
<code><a href="getBio.html">getBio</a>()</code>,
<code><a href="getCDS.html">getCDS</a>()</code>,
<code><a href="getCollection.html">getCollection</a>()</code>,
<code><a href="getGFF.html">getGFF</a>()</code>,
<code><a href="getGenome.html">getGenome</a>()</code>,
<code><a href="getRNA.html">getRNA</a>()</code></p>
<p>Other proteome: 
<code><a href="getProteomeSet.html">getProteomeSet</a>()</code>,
<code><a href="read_proteome.html">read_proteome</a>()</code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="co"># download the proteome of Arabidopsis thaliana from NCBI RefSeq</span></span></span>
<span class="r-in"><span><span class="co"># and store the corresponding proteome file in '_ncbi_downloads/refseq/proteomes'</span></span></span>
<span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu">getProteome</span><span class="op">(</span> db       <span class="op">=</span> <span class="st">"refseq"</span>,</span></span>
<span class="r-in"><span>             organism <span class="op">=</span> <span class="st">"Arabidopsis thaliana"</span>,</span></span>
<span class="r-in"><span>             path     <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"_ncbi_downloads"</span>,<span class="st">"refseq"</span>,<span class="st">"proteomes"</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># import proteome into R session</span></span></span>
<span class="r-in"><span><span class="va">Ath_proteome</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_proteome.html">read_proteome</a></span><span class="op">(</span><span class="va">file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># download the proteome of Arabidopsis thaliana from NCBI Genbank</span></span></span>
<span class="r-in"><span><span class="co"># and store the corresponding proteome file in '_ncbi_downloads/genbank/proteomes'</span></span></span>
<span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu">getProteome</span><span class="op">(</span> db       <span class="op">=</span> <span class="st">"genbank"</span>,</span></span>
<span class="r-in"><span>             organism <span class="op">=</span> <span class="st">"Arabidopsis thaliana"</span>,</span></span>
<span class="r-in"><span>             path     <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"_ncbi_downloads"</span>,<span class="st">"genbank"</span>,<span class="st">"proteomes"</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># import proteome into R session</span></span></span>
<span class="r-in"><span><span class="va">Ath_proteome</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_proteome.html">read_proteome</a></span><span class="op">(</span><span class="va">file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># and store the corresponding proteome file in '_downloads/uniprot/proteomes'</span></span></span>
<span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu">getProteome</span><span class="op">(</span> db       <span class="op">=</span> <span class="st">"uniprot"</span>,</span></span>
<span class="r-in"><span>             organism <span class="op">=</span> <span class="st">"Arabidopsis thaliana"</span>,</span></span>
<span class="r-in"><span>             path     <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"_downloads"</span>,<span class="st">"uniprot"</span>,<span class="st">"proteomes"</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># import proteome into R session</span></span></span>
<span class="r-in"><span><span class="va">Ath_proteome</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_proteome.html">read_proteome</a></span><span class="op">(</span><span class="va">file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># download the proteome of Arabidopsis thaliana from ENSEMBL</span></span></span>
<span class="r-in"><span><span class="co"># and store the corresponding proteome file in '_downloads/ensembl/proteomes'</span></span></span>
<span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu">getProteome</span><span class="op">(</span> db       <span class="op">=</span> <span class="st">"ensembl"</span>,</span></span>
<span class="r-in"><span>             organism <span class="op">=</span> <span class="st">"Arabidopsis thaliana"</span>,</span></span>
<span class="r-in"><span>             path     <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"_downloads"</span>,<span class="st">"ensembl"</span>,<span class="st">"proteomes"</span><span class="op">)</span> <span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># import proteome into R session</span></span></span>
<span class="r-in"><span><span class="va">Ath_proteome</span> <span class="op">&lt;-</span> <span class="fu"><a href="read_proteome.html">read_proteome</a></span><span class="op">(</span><span class="va">file_path</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

