<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perform Meta-Genome Retrieval — meta.retrieval • biomartr</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Perform Meta-Genome Retrieval — meta.retrieval"><meta property="og:description" content="Download genomes, proteomes, cds, gff, rna, or assembly stats
files of all species within a kingdom of life."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomartr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.11</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/BioMart_Examples.html">Ensembl BioMart Examples</a>
    </li>
    <li>
      <a href="../articles/Database_Retrieval.html">NCBI Database Retrieval</a>
    </li>
    <li>
      <a href="../articles/Functional_Annotation.html">Functional Annotation with biomartr</a>
    </li>
    <li>
      <a href="../articles/MetaGenome_Retrieval.html">Meta-Genome Retrieval</a>
    </li>
    <li>
      <a href="../articles/Sequence_Retrieval.html">Sequence Retrieval</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ropensci/biomartr/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform Meta-Genome Retrieval</h1>
    <small class="dont-index">Source: <a href="https://github.com/ropensci/biomartr/blob/HEAD/R/meta.retrieval.R" class="external-link"><code>R/meta.retrieval.R</code></a></small>
    <div class="hidden name"><code>meta.retrieval.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Download genomes, proteomes, cds, gff, rna, or assembly stats
files of all species within a kingdom of life.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">meta.retrieval</span><span class="op">(</span></span>
<span>  db <span class="op">=</span> <span class="st">"refseq"</span>,</span>
<span>  <span class="va">kingdom</span>,</span>
<span>  group <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  type <span class="op">=</span> <span class="st">"genome"</span>,</span>
<span>  restart_at_last <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  combine <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-db">db<a class="anchor" aria-label="anchor" href="#arg-db"></a></dt>
<dd><p>a character string specifying the database from which the genome
shall be retrieved:</p>
<ul><li><p><code>db = "refseq"</code></p></li>
<li><p><code>db = "genbank"</code></p></li>
<li><p><code>db = "emsembl"</code></p></li>
</ul></dd>


<dt id="arg-kingdom">kingdom<a class="anchor" aria-label="anchor" href="#arg-kingdom"></a></dt>
<dd><p>a character string specifying the kingdom of the organisms
of interest, e.g.</p>
<ul><li><p>For <code>NCBI RefSeq</code>:</p><ul><li><p><code>kingdom = "archaea"</code></p></li>
<li><p><code>kingdom = "bacteria"</code></p></li>
<li><p><code>kingdom = "fungi"</code></p></li>
<li><p><code>kingdom = "invertebrate"</code></p></li>
<li><p><code>kingdom = "plant"</code></p></li>
<li><p><code>kingdom = "protozoa"</code></p></li>
<li><p><code>kingdom = "viral"</code></p></li>
<li><p><code>kingdom = "vertebrate_mammalian"</code></p></li>
<li><p><code>kingdom = "vertebrate_other"</code></p></li>
</ul></li>
<li><p>For <code>NCBI Genbank</code>:</p><ul><li><p><code>kingdom = "archaea"</code></p></li>
<li><p><code>kingdom = "bacteria"</code></p></li>
<li><p><code>kingdom = "fungi"</code></p></li>
<li><p><code>kingdom = "invertebrate"</code></p></li>
<li><p><code>kingdom = "plant"</code></p></li>
<li><p><code>kingdom = "protozoa"</code></p></li>
<li><p><code>kingdom = "vertebrate_mammalian"</code></p></li>
<li><p><code>kingdom = "vertebrate_other"</code></p></li>
</ul></li>
<li><p>For <code>ENSEMBL</code>:</p><ul><li><p><code>kingdom = "EnsemblVertebrates"</code></p></li>
<li><p><code>kingdom = "EnsemblPlants"</code></p></li>
<li><p><code>kingdom = "EnsemblFungi"</code></p></li>
<li><p><code>kingdom = "EnsemblMetazoa"</code></p></li>
<li><p><code>kingdom = "EnsemblBacteria"</code></p></li>
<li><p><code>kingdom = "EnsemblProtists"</code></p></li>
</ul></li>
</ul><p>Available kingdoms can be retrieved with <code><a href="getKingdoms.html">getKingdoms</a></code>.</p></dd>


<dt id="arg-group">group<a class="anchor" aria-label="anchor" href="#arg-group"></a></dt>
<dd><p>only species belonging to this subgroup will be downloaded.
Groups can be retrieved with <code><a href="getGroups.html">getGroups</a></code>.</p></dd>


<dt id="arg-type">type<a class="anchor" aria-label="anchor" href="#arg-type"></a></dt>
<dd><p>type of sequences that shall be retrieved. Options are:</p>
<ul><li><p><code>type = "genome"</code> :
 (for genome assembly retrieval; see also <code><a href="getGenome.html">getGenome</a></code>),</p></li>
<li><p><code>type = "proteome"</code> :
 (for proteome retrieval; see also <code><a href="getProteome.html">getProteome</a></code>),</p></li>
<li><p><code>type = "cds"</code> :
 (for coding sequence retrieval; see also <code><a href="getCDS.html">getCDS</a></code>),</p></li>
<li><p><code>type = "gff"</code> :
(for annotation file retrieval in gff format; see also <code><a href="getGFF.html">getGFF</a></code>),</p></li>
<li><p><code>type = "gtf"</code> :
(for annotation file retrieval in gtf format (only for ensembl and
 ensemblgenomes); see also <code><a href="getGTF.html">getGTF</a></code>)</p></li>
<li><p><code>type = "rna"</code> :
 (for RNA file retrieval in fasta format; see also <code><a href="getRNA.html">getRNA</a></code>),</p></li>
<li><p><code>type = "rm"</code> :
 (for Repeat Masker output file retrieval; see also
 <code><a href="getRepeatMasker.html">getRepeatMasker</a></code>),</p></li>
<li><p><code>type = "assemblystats"</code> :
 (for genome assembly quality stats file retrieval;
 see also <code><a href="getAssemblyStats.html">getAssemblyStats</a></code>).</p></li>
</ul></dd>


<dt id="arg-restart-at-last">restart_at_last<a class="anchor" aria-label="anchor" href="#arg-restart-at-last"></a></dt>
<dd><p>a logical value indicating whether or not <code>meta.retrieval</code> should pick up at the last species when re-running the function.</p><ul><li><p>If <code>restart_at_last = TRUE</code> (Default) then <code>meta.retrieval</code> will skip all organisms that are already present in the folder
and will start downloading all remaining species. However, this way <code>meta.wretrieval</code> will not be able to check whether
already downloaded organism files are corrupted or not by checking the md5 checksum.</p></li>
<li><p>If <code>restart_at_last = FALSE</code> then <code>meta.retrieval</code> will start from the beginning and crawl through already downloaded
organism files and check whether already downloaded organism files are corrupted or not by checking the md5 checksum.
After checking existing files the function will start downloading all remaining organisms.</p></li>
</ul></dd>


<dt id="arg-reference">reference<a class="anchor" aria-label="anchor" href="#arg-reference"></a></dt>
<dd><p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database
as either a reference genome or a representative genome. Options are:</p><ul><li><p><code>reference = FALSE</code> (Default): all organisms (reference, representative, and non-representative genomes) are downloaded.</p></li>
<li><p><code>reference = TRUE</code>: organisms that are downloaded must be either a reference or representative genome. Thus, most genomes which are usually non-reference genomes
will not be downloaded.</p></li>
</ul></dd>


<dt id="arg-combine">combine<a class="anchor" aria-label="anchor" href="#arg-combine"></a></dt>
<dd><p>just in case <code>type = "assemblystats"</code> is specified, shall
assemby stats of individual species be imported and combined to a
<code><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></code>?</p></dd>


<dt id="arg-path">path<a class="anchor" aria-label="anchor" href="#arg-path"></a></dt>
<dd><p>path to the folder in which downloaded genomes shall be stored.
By default the kingdom name is used to name the output folder.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a character vector storing the file paths of the retrieved files.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function aims to perform bulk retrieval of the genomes,
proteomes, cds, etc. of species that belong to the same kingdom of life or
to the same subgroup.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>Other meta_retrival:
<code><a href="meta.retrieval.all.html">meta.retrieval.all</a>()</code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># get all available kingdoms for refseq</span></span></span>
<span class="r-in"><span><span class="fu"><a href="getKingdoms.html">getKingdoms</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># download all vertebrate genomes from refseq</span></span></span>
<span class="r-in"><span><span class="fu">meta.retrieval</span><span class="op">(</span>kingdom <span class="op">=</span> <span class="st">"vertebrate_mammalian"</span>,</span></span>
<span class="r-in"><span>               db <span class="op">=</span> <span class="st">"refseq"</span>,</span></span>
<span class="r-in"><span>               type <span class="op">=</span> <span class="st">"genome"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get all available kingdoms for genbank</span></span></span>
<span class="r-in"><span><span class="fu"><a href="getKingdoms.html">getKingdoms</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"genbank"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># download all vertebrate genomes from genbank</span></span></span>
<span class="r-in"><span><span class="fu">meta.retrieval</span><span class="op">(</span>kingdom <span class="op">=</span> <span class="st">"vertebrate_mammalian"</span>,</span></span>
<span class="r-in"><span>               db <span class="op">=</span> <span class="st">"genbank"</span>,</span></span>
<span class="r-in"><span>               type <span class="op">=</span> <span class="st">"genome"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># In case users do not wish to retrieve genomes from an entire kingdom,</span></span></span>
<span class="r-in"><span><span class="co"># but rather from a subgoup (e.g. from species belonging to the</span></span></span>
<span class="r-in"><span><span class="co"># Gammaproteobacteria class, a subgroup of the bacteria kingdom),</span></span></span>
<span class="r-in"><span><span class="co"># they can use the following workflow"</span></span></span>
<span class="r-in"><span><span class="co"># First, users can again consult the getKingdoms() function to retrieve</span></span></span>
<span class="r-in"><span><span class="co"># kingdom information.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="getKingdoms.html">getKingdoms</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># In this example, we will choose the bacteria kingdom.</span></span></span>
<span class="r-in"><span><span class="co"># Now, the getGroups() function allows users to obtain available</span></span></span>
<span class="r-in"><span><span class="co"># subgroups of the bacteria kingdom.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="getGroups.html">getGroups</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"refseq"</span>, kingdom <span class="op">=</span> <span class="st">"bacteria"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Now we choose the group Gammaproteobacteria and specify</span></span></span>
<span class="r-in"><span><span class="co"># the group argument in the meta.retrieval() function</span></span></span>
<span class="r-in"><span><span class="fu">meta.retrieval</span><span class="op">(</span>kingdom <span class="op">=</span> <span class="st">"bacteria"</span>,</span></span>
<span class="r-in"><span>   roup <span class="op">=</span> <span class="st">"Gammaproteobacteria"</span>,</span></span>
<span class="r-in"><span>   db <span class="op">=</span> <span class="st">"refseq"</span>,</span></span>
<span class="r-in"><span>   type <span class="op">=</span> <span class="st">"genome"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Haakon Tjeldnes.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

